---
title: Cis-regulatory mutations associate with transcriptional and post-transcriptional deregulation of the gene regulatory program in cancers
author:
- Jaime A. Castro-Mondragon,
- Miriam Ragle Aure,
- Ole Christian Lingjærde,
- Anita Langerød,
- John W. M. Martens,
- Anne-Lise Børresen-Dale,
- Vessela Kristensen,
- and Anthony Mathelier
subtitle: 'Supplemental Material'
date: 'Last update: `r Sys.Date()`'
geometry: margin=1.5cm
output: 
  pdf_document
bibliography: References.bib
biblio-style: apalike
link-citations: yes
documentclass: article
#classoption: openany
header-includes: 
  \usepackage{float} \floatplacement{figure}{H} 
  \newcommand{\beginsupplement}{\setcounter{table}{0}
  \renewcommand{\thetable}{S\arabic{table}} \setcounter{figure}{0}
  \renewcommand{\thefigure}{S\arabic{figure}}}
  \usepackage{fullpage}
  \usepackage{pdflscape}
  \usepackage{inputenc}
---

\beginsupplement


```{r setup, include=FALSE, cache=TRUE, echo=FALSE, eval=TRUE}
knitr::opts_chunk$set(echo = TRUE)
suppressPackageStartupMessages(library(bookdown))
suppressPackageStartupMessages(library(ComplexHeatmap))
suppressPackageStartupMessages(library(cowplot))
suppressPackageStartupMessages(library(ggplot2))
suppressPackageStartupMessages(library(UpSetR))
suppressPackageStartupMessages(library(scales))
suppressPackageStartupMessages(library(emojifont))
suppressPackageStartupMessages(library(Cairo))
suppressPackageStartupMessages(library(leaflet))
suppressPackageStartupMessages(library(ggplotify))
suppressPackageStartupMessages(library(ggpubr))
suppressPackageStartupMessages(library(ggsignif))
suppressPackageStartupMessages(library(ggnetwork))

pdf.options(encoding = 'ISOLatin2')
options(device = "cairo_pdf")


## Custom
blank_theme <- theme_minimal() +
  theme(
    axis.title.x = element_blank(),
    axis.title.y = element_blank(),
    panel.border = element_blank(),
    panel.grid=element_blank(),
    axis.ticks = element_blank(),
    plot.title=element_text(size=14, face="bold")
  )

add_space <-   theme(
    panel.background = element_rect(fill = "white", colour = "white"),
    plot.margin = margin(0.5, 0.5, 0.5, 0.5, "cm"),
    plot.background = element_rect(
      fill = "#FFFEFD",
      colour = "#FFFEFD",
      size = 0.001
    )
  )


add_space_gg <-   theme(
    panel.background = element_rect(fill = "white"),
    plot.margin = margin(0.5, 0.5, 0.5, 0.5, "cm"),
    plot.background = element_rect(
      fill = "#FFFEFD",
      colour = "white",
      size = 1
    ))


agg.net.fun <- function(df           = NULL,
                        project.name = NULL,
                        clusters     = NULL) {

  ggplot(df, aes(x = x, y = y, xend = xend, yend = yend, color = Nb_samples, fill = TF_gene)) +
        geom_edges(color = "grey50") +
        geom_nodelabel_repel(aes(label = vertex.names),
                             fontface = "plain", box.padding = unit(1, "lines"), color = "black", force = 0.85, size = 6) +
        scale_fill_manual(values = c("#fdbb84", "white"),
                          breaks = c("TF", "non_TF")) + ##
        geom_edges(arrow = arrow(length = unit(7, "pt"), type = "closed")) +
        geom_nodes(size = 8) +
        scale_color_distiller(palette = "PuBu", direction = +1, limits = c(1 ,max(df$Nb_samples)), name = "Samples    ") + 
        theme_blank() +
        labs(title = paste("Aggregated network ", project.name, " - Genes: ", sum(clusters$csize), " - Subgraphs: ", clusters$no)) +
        guides(fill = FALSE) +
        theme(plot.title = element_text(hjust = 0.5, size=12),
              legend.text=element_text(size=10),
              legend.title=element_text(size=10),
              legend.position = "bottom",
              legend.box = "vertical") +
        guides(colour = guide_colourbar(barwidth = 7.5, barheight = 2))
}


# Laptop -> Biotin
# rsync -rptuvl /home/jamondra/Documents/PostDoc/Mathelier_lab/Manuscript/Supp_Material jamondra@biotin2.hpc.uio.no:/storage/mathelierarea/processed/jamondra/Projects/dysmir/Manuscript

# Biotin -> Laptop
# rsync -rptuvl jamondra@biotin2.hpc.uio.no:/storage/mathelierarea/processed/jamondra/Projects/dysmir/Manuscript/Supp_Material /home/jamondra/Documents/PostDoc/Mathelier_lab/Manuscript

```



\pagebreak
```{r Number_of_Mutations, include=TRUE, cache=TRUE, echo=FALSE, eval=TRUE, fig.cap="A) Number of mutations per sample per cohort. B) Number of cis-regulatory mutations (i.e. within TFBSs) per sample per cohort.", fig.height = 8, fig.width = 6, fig.align = "center"}

load("Mutation_frequency/Nb_mutated_nt_within_TFBSs_across_cohorts.RData")
load("Mutation_frequency/Nb_mutations_per_cohort_dotplot.RData")

wgs.mut.summary.plot <-
  wgs.mut.summary.plot +
  theme(
    panel.background = element_rect(fill = "white"),
    plot.margin = margin(0.5, 0.5, 0.5, 0.5, "cm"),
    plot.background = element_rect(
      fill = "#fefefe",
      colour = "white",
      size = 1
    )
  ) +
  labs(title = "Number of mutations")


mutated.tfbs.cohorts <-
  mutated.tfbs.cohorts +
  theme(
    panel.background = element_rect(fill = "white"),
    plot.margin = margin(0.5, 0.5, 0.5, 0.5, "cm"),
    plot.background = element_rect(
      fill = "#fefefe",
      colour = "white",
      size = 1
    )
  ) +
  labs(title = "Number of mutations in TFBSs")

suppressWarnings(
plot_grid(wgs.mut.summary.plot, mutated.tfbs.cohorts, ncol = 1, labels = c('A)', 'B)'), label_size = 17, align = "v", hjust = -0, vjust = 2.9)
)
```





\pagebreak
```{r Mutation_fraction_in_TFBSs, include=TRUE, echo=FALSE, eval=TRUE, fig.cap="Fraction of somatic mutations per sample per cohort categorized as cis-regulatory mutations (i.e. within TFBSs).", fig.height = 5, fig.width = 7, fig.align = "center"}

library(ggplot2)
load("Mutation_frequency/Fraction_of_somatic_mutations_in_TFBSs.RData")

Fraction.mut.in.tfbs.gg <- Fraction.mut.in.tfbs.gg +
                            theme_minimal() +
                            theme(
                              panel.background = element_rect(fill = "white"),
                              # plot.margin = margin(0.5, 0.5, 0.5, 0.5, "cm"),
                              plot.background = element_rect(
                                fill = "#fefefe",
                                colour = "white",
                                size = 1
                              )
                            )

suppressWarnings(print(Fraction.mut.in.tfbs.gg))

```





\pagebreak
```{r Mutation_rates_paired_1, cache=TRUE, include=TRUE, echo=FALSE, eval=TRUE, fig.cap="Paired comparisons (wilcoxon tests) of mutation rates in A) BRCA samples (n = 92), B) LIHC (n = 50), C) UCEC (n = 48), and D) HNSC (n = 43) at TFBS, exonic, flanking, and random regions.", fig.height = 19, fig.width = 15, fig.align = "center"}

load("Mutation_rate/Paired_Comparison_Mutation_rates_BRCA-US.RData")
mut.rate.distrib.comp.gg.brca <- mut.rate.distrib.comp.gg + add_space

load("Mutation_rate/Paired_Comparison_Mutation_rates_LIHC-US.RData")
mut.rate.distrib.comp.gg.lihc <- mut.rate.distrib.comp.gg + add_space

load("Mutation_rate/Paired_Comparison_Mutation_rates_UCEC-US.RData")
mut.rate.distrib.comp.gg.ucec <- mut.rate.distrib.comp.gg + add_space

load("Mutation_rate/Paired_Comparison_Mutation_rates_HNSC-US.RData")
mut.rate.distrib.comp.gg.hnsc <- mut.rate.distrib.comp.gg + add_space

suppressMessages(suppressWarnings(plot_grid(mut.rate.distrib.comp.gg.brca, 
                                            mut.rate.distrib.comp.gg.lihc,
                                            mut.rate.distrib.comp.gg.ucec,
                                            mut.rate.distrib.comp.gg.hnsc,
                                            ncol = 1,
                                            nrow = 4,
                                            labels = c('A)', 'B)', 'C)', 'D)'),
                                            label_size = 17,
                                            align = "hv",
                                            rel_heights = c(0.8),
                                            vjust = 1)))
```





\pagebreak
```{r Mutation_rates_paired_2, cache=TRUE, include=TRUE, echo=FALSE, eval=TRUE, fig.cap="Paired comparisons (wilcoxon tests) of mutation rates in A) STAD samples (n = 35), B) LUAD (n = 37), and D) LUSC (n = 42) at TFBS, exonic, flanking, and random regions.", fig.height = 19, fig.width = 15, fig.align = "center"}

load("Mutation_rate/Paired_Comparison_Mutation_rates_STAD-US.RData")
mut.rate.distrib.comp.gg.stad <- mut.rate.distrib.comp.gg + add_space

load("Mutation_rate/Paired_Comparison_Mutation_rates_LUAD-US.RData")
mut.rate.distrib.comp.gg.luad <- mut.rate.distrib.comp.gg + add_space

load("Mutation_rate/Paired_Comparison_Mutation_rates_LUSC-US.RData")
mut.rate.distrib.comp.gg.lusc <- mut.rate.distrib.comp.gg + add_space

suppressMessages(suppressWarnings(plot_grid(mut.rate.distrib.comp.gg.stad, 
                                            mut.rate.distrib.comp.gg.luad,
                                            mut.rate.distrib.comp.gg.lusc,
                                            ncol = 1,
                                            nrow = 3,
                                            labels = c('A)', 'B)', 'C)'),
                                            label_size = 17,
                                            align = "hv",
                                            rel_heights = c(0.9),
                                            vjust = 1)))
```





\pagebreak
```{r TFBS_vs_Exonic_mutation_rate_BRCA, cache=TRUE, include=TRUE, echo=FALSE, eval=TRUE, fig.cap="TFBS versus exonic mutation rates in TCGA cohorts. Both axis are -log$_{10}$ converted. The diagonal line indicates identical mutation rates in TFBS and exonic regions. The blue line represents a linear regression model best fitting the points with a 95\\% confidence interval (grey zone).", fig.height = 12, fig.width = 12, fig.align = "center"}


load("Mutation_rate/TFBS_exonic_mut_rates_facet.Rdata")
TFBS.exonic.mut.rate.facet

```





\pagebreak
```{r TFBS_flanks, cache=TRUE, include=TRUE, echo=FALSE, eval=TRUE, fig.cap="Mutation rates in TFBSs and their flanking regions. Left: ±50 bp, center: ±250 bp, right: ±500 bp.", fig.height = 12.5, fig.width = 11, fig.align = "center"}

load("Mutation_rate/Mut_rates_TFBS_vs_flanks_facet_all_cohorts.RData")
tfbs.vs.flanks.gg.all + add_space

```





\pagebreak
```{r Exon_flanks, cache=TRUE, include=TRUE, echo=FALSE, eval=TRUE, fig.cap="Mutation rates in exons and their flanking regions. Left: ±50 bp, center: ±250 bp, right: ±500 bp.", fig.height = 12.5, fig.width = 11, fig.align = "center"}

load("Mutation_rate/Mut_rates_Exons_vs_flanks_facet_all_cohorts.RData")
exons.vs.flanks.gg.all + add_space

```





\pagebreak
```{r GeneHancer_associations, cache=TRUE, include=TRUE, echo=FALSE, eval=TRUE, fig.cap="TFBS-gene associations. When a TFBS (purple box) lies within an GeneHancer-annotated cis-regulatory element (CRE; pink box), the TFBS is linked to its target genes according to GeneHancer, (see TFBS$_B$ and TFBS$_C$). When a TFBS is outside an annotated CRE (see TFBS$_A$, TFBS$_D$, and TFBS$_E$), it is associated with the closest TSS.", out.width = "500px", fig.align = "center"}

knitr::include_graphics("GeneHancer/TFBS_CRE_association-1.png", dpi = 100)
```





\pagebreak
```{r xseq_model_explained, cache=TRUE, include=TRUE, echo=FALSE, eval=TRUE, fig.cap="Likelihood association of mutations with gene expression dysregulation in xseq. Example with a network of target genes for a given miRNA. A) Gene regulatory status. For each target $i$ connected to the miRNA gene $X$ associated with a cis-regulatory mutation, xseq decomposes the expression distribution of $i$ (using all samples in the cohort) as a 3-component mixture model (down, neutral, and up). Next, xseq computes the posterior probability of observing the expression of $i$ in the mutated sample $S$ as belonging to the down, neutral, and up components. The component with the highest posterior probability provides information about the regulatory status of $i$ (down, neutral, or up). This step is repeated for each gene in the network and for each sample in the cohort. B) The sample-specific network dysregulation probability is calculated from the down and up regulation posterior probabilities of all the genes in the network for a given samples where the mutation was observed. C) The posterior probabilities of sample-specific dysregulation are used to compute the posterior probability of association between a cis-regulatory mutated miRNA and its target network dysregulation across a cohort. For a detailed explanation of the parameters and the graphical model, see the xseq publication by Ding et al., Nature Communications, 2015.", fig.align = "center", out.width = "500px"}

knitr::include_graphics("Model/Xseq_explained.png", dpi = 100)
```





\pagebreak
```{r Predicted_PCG_1, cache=TRUE, include=TRUE, echo=FALSE, eval=TRUE, fig.cap="Protein-coding genes predicted in BRCA-US, HNSC-US, LIHC-US, and LUSC-US TCGA cohorts. Predictions were obtained applying the xseq tool when considering protein-coding genes mutated through either LoF (red triangles) or cis-regulatory (TFBS; green triangles) mutations, independently. Cancer-associated genes (red stars) and TFs (blue stars) are highlighted and hypergeometric tests p-values for enrichment are provided in the legend (Material and methods).", out.width = "650px", out.height="550px", fig.align = "center"}

knitr::include_graphics("Highlighted_genes_pancancer/xseq_trans_pancancer_heatmap_PCG_sep_by_cohort_P1.jpeg", dpi = 100)
```





\pagebreak
```{r Predicted_PCG_2, cache=TRUE, include=TRUE, echo=FALSE, eval=TRUE, fig.cap="Protein-coding genes predicted in LUAD-US, STAD-US, and UCEC-US TCGA cohorts. Predictions were obtained applying the xseq tool when considering protein-coding genes mutated through either LoF (red triangles) or cis-regulatory (TFBS; green triangles) mutations, independently. Cancer-associated genes (red stars) and TFs (blue stars) are highlighted and hypergeometric tests p-values for enrichment are provided in the legend (Material and methods).", out.width = "700px", out.height="575px", fig.align = "center"}

knitr::include_graphics("Highlighted_genes_pancancer/xseq_trans_pancancer_heatmap_PCG_sep_by_cohort_P2.jpeg", dpi = 100)
```





\pagebreak
```{r TFBS_vs_Exonic_mutations, cache=TRUE, include=TRUE, echo=FALSE, eval=TRUE, fig.cap="Protein-coding genes predicted through cis-regulatory mutations (i.e. within TFBSs) are generally not mutated in their exons in the same patients. A) For each cohort, number of samples where the predicted protein-coding genes through cis-regulatory mutations (TFBSs) also contain a LoF mutation. B) Same as in A) but providing the corresponding number of predicted protein-coding genes per cohort for each category.", fig.height = 15, fig.width = 13, fig.align = "center"}

load("Nb_TFBS_vs_Exonic_mutations/Analysis_samples_w_TFBS_Exonic_mutations.rdata")


## The original colors in the legend are inverted when the plots are ecported as Rdata, therefore we have to reset the color scale
suppressMessages(barplots$A     <- barplots$A + scale_fill_manual(values = c(TFBS_and_Exonic = "#bf812d", TFBS = "#35978f"), labels = c("TFBS only", "TFBS & Exon")))

suppressMessages(barplots$extra <- barplots$extra + scale_fill_manual(values = c(TFBS_and_Exonic = "#bf812d", TFBS = "#35978f"), labels = c("TFBS only", "TFBS & Exon")))

suppressMessages(barplots$B     <- barplots$B + scale_fill_manual(values = c(TFBS_and_Exonic = "#bf812d", TFBS = "#35978f"), labels = c("TFBS only", "TFBS & Exon")))
  
font.size <- 100

tfbs.vs.exon.mutations <- plot_grid(barplots$A, barplots$extra, barplots$B,
                                    nrow = 3, rel_heights = c(1, 0.001, 1),
                                    labels = c("A)", "", "B)"),
                                    label_size = 25)
tfbs.vs.exon.mutations <- tfbs.vs.exon.mutations + add_space


suppressMessages(tfbs.vs.exon.mutations )
```





\pagebreak
```{r Aggregated_networks_1, cache=TRUE, include=TRUE, echo=FALSE, eval=TRUE, fig.cap="Network of all predicted genes in A) BRCA, B) HNSC, C) LIHC, D) LUAD, E) LUSC, and F) STAD cohorts. The node colors indicate the number of samples where the gene was predicted. TFs genes are highlighted with orange background.", fig.height = 30, fig.width = 25, fig.align = "center"}

cohorts <- c("BRCA", "HNSC", "LIHC", "LUAD", "LUSC", "STAD")

net.list <- vector(mode = "list", length = length(cohorts))
for (co in cohorts) {
  
  load(paste0("Aggregated_networks/", co, "/Aggregated_network_", co, "-US.Rdata"))
  agg.net <- agg.net.fun(df           = list.agg.net$DF,
                         project.name = list.agg.net$Project,
                         clusters     = list.agg.net$Clusters)  + add_space
  
  net.list[[co]] <- agg.net
  
}




suppressMessages(plot_grid(net.list[["BRCA"]],
                           net.list[["HNSC"]],

                           net.list[["LIHC"]],
                           net.list[["LUAD"]],

                           net.list[["LUSC"]],
                           net.list[["STAD"]],
                           nrow = 3, labels = c('A)', 'B)', 'C)', 'D)', 'E)', 'F)'),
                           label_size = 35, vjust = 1, hjust = 0))
```



<!-- \pagebreak -->
<!-- ```{r Dysregulation_HeatMaps_PCGs, cache=TRUE, include=TRUE, echo=FALSE, eval=TRUE, fig.cap="Examples of dysregulated networks for predicted protein-coding driver genes through cis-regulatory mutations: GATA3 in BRCA-US (left), GATA3 in LUAD-US (center), and LA2G4A in BRCA-US (right). Each heatmap column corresponds to a sample where TFBSs associated to the predicted drivers are found with a somatic mutation. Each row corresponds to dysregulated genes in the network of the predicted driver. The color scale represents the gene regulatory status (red: up-regulation; blue: down-regulation). The horizontal bars on top of the heatmaps show the patient-specific driver posterior probability; the darker the higher the probability. Note patient stratification based on this probability.", fig.height = 28, fig.width = 25, fig.align = "center"} -->

<!-- ## Load the heatmap objects -->
<!-- load("Dysreg_heatmaps_examples/PCG/Dysregulation_heatmaps_BRCA-US.Rdata") ## all.heatmaps.list -->
<!-- all.heatmaps.list.brca <- all.heatmaps.list -->

<!-- load("Dysreg_heatmaps_examples/PCG/Dysregulation_heatmaps_LUAD-US.Rdata") ## all.heatmaps.list -->
<!-- all.heatmaps.list.luad <- all.heatmaps.list -->

<!-- load("Dysreg_heatmaps_examples/PCG/Dysregulation_heatmaps_LUSC-US.Rdata") ## all.heatmaps.list -->
<!-- all.heatmaps.list.lusc <- all.heatmaps.list -->

<!-- ht1 <- all.heatmaps.list.brca[["GATA3_LoF"]] -->
<!-- ht2 <- all.heatmaps.list.luad[["GATA3_TFBS"]] -->
<!-- ht3 <- all.heatmaps.list.brca[["PLA2G4A_TFBS"]] -->
<!-- # ht3 <- all.heatmaps.list.lusc[["PLA2G4A_TFBS"]] -->


<!-- ht11 <- as.ggplot(grid.grabExpr(draw(ht1))) + add_space -->
<!-- ht22 <- as.ggplot(grid.grabExpr(draw(ht2))) + add_space -->
<!-- ht33 <- as.ggplot(grid.grabExpr(draw(ht3))) + add_space -->

<!-- plot_grid(ht11, ht22, ht33, -->
<!--           nrow =  1) -->

<!-- rm(ht1) -->
<!-- rm(ht2) -->
<!-- rm(ht3) -->
<!-- rm(ht11) -->
<!-- rm(ht22) -->
<!-- rm(ht33) -->
<!-- rm(all.heatmaps.list) -->
<!-- rm(all.heatmaps.list.brca) -->
<!-- rm(all.heatmaps.list.luad) -->
<!-- rm(all.heatmaps.list.lusc) -->
<!-- ``` -->



\pagebreak
```{r FEA_GO_BP, cache=TRUE, include=TRUE, echo=FALSE, eval=TRUE, fig.cap="Functional enrichment (GO Biological Process 2021) analysis considering dysregulated genes in the networks of predicted miRNA and protein-coding genes. The heatmaps represent the significance (-log$_{10}$(P-value)) of the most enriched terms (rows) associated to the dysregulated genes found on the seven analyzed cohorts (columns) when considering protein-coding genes with LoF mutations (PCGs::LoF; top-left), protein-coding genes with cis-regulatory mutations (PCG::TFBS; top-right), and miRNAs with cis-regulatory mutations (miRNAs::TFBS; bottom-right). The terms are ordered by their mean ranks across the cohorts.", fig.height = 110, fig.width = 97, fig.align = "center"}

types <- c("PCG::TFBS",
           "premiRNA::TFBS",
           "PCG::LoF")

load("FEA_heatmaps/GSEA_enrichment_pancancer_heatmap_ordered_labels.RData")  ## heatmap.GSEA.axis.list


db <- "GO_Biological_Process_2021"
font.size <- 70
# co <- "PCG::TFBS"
heatmap.GSEA.list <- list()
for(co in types){

  fea.df <- heatmap.GSEA.axis.list[[db]][[co]][["df"]]

  cohorts.sorted <- heatmap.GSEA.axis.list[[db]][[co]][["x_axis"]]
  terms.sorted <- heatmap.GSEA.axis.list[[db]][[co]][["y_axis"]]
  ti = co
  if(co=="premiRNA::TFBS")
    ti = "miRNA::TFBS"

  heatmap.GSEA.list[[co]] <-
    ggplot(fea.df, aes(x = Cohort, y = Term, fill = Significance)) +
        geom_tile(width = 0.95, height = 0.95) +
        # coord_equal() +
        theme_bw() +
        theme(legend.position = "bottom", legend.title=element_text(size=font.size)) +
        scale_fill_viridis_c(option = "A",direction = -1, breaks = floor(seq(0, max(fea.df$Significance), length.out = 4)), limits = c(0, max(fea.df$Significance))) + # limits = c(0, 75)
        # scale_fill_distiller(palette = "PuRd", direction = +1, breaks = gg.breaks) +
        theme(axis.text.x = element_text(angle = 45, hjust = 1, vjust=1, size=font.size, colour= "black"),
              axis.text.y = element_text(size=font.size, colour = "black"),
              axis.title.y = element_text(size=font.size,colour = "black",vjust=0.12),
              axis.title.x = element_text(size=font.size,colour = "black",vjust=0.12),
              legend.text=element_text(size=font.size),
              plot.title = element_text(size=font.size)) +
        scale_x_discrete(limits = cohorts.sorted) +
        scale_y_discrete(limits = terms.sorted) +
        labs(title = paste0(db, "\n", ti), x = "", y = "") +
        guides(fill = guide_colourbar(barwidth = 23, barheight = 6)) +
        guides(color = guide_legend(title.position = "top",
                                    title.hjust = 0.5,
                                    label.position = "left")) +
    ## Add a border to avoid trunked axis legends
      theme(
        panel.background = element_rect(fill = "white"),
        plot.margin = margin(0.25, 0.25, 0.25, 0.25, "cm"),
        plot.background = element_rect(
          fill = "#fefefe",
          colour = "white",
          size = 1))
}


enriched.terms.gg <- plot_grid(heatmap.GSEA.list[["PCG::LoF"]],
                               heatmap.GSEA.list[["PCG::TFBS"]], NULL,
                               heatmap.GSEA.list[["premiRNA::TFBS"]],
                               ncol = 2, align = "hv", labels = c('', '', ''), label_size = 30,
                               label_colour = "black", hjust = -2, vjust = 3,
                               rel_heights = c(1, 1, 1,1), rel_widths = c(1, 1, 1,1))

## Combine plots
enriched.terms.gg

rm(heatmap.GSEA.list)
rm(enriched.terms.gg)
```



\pagebreak
```{r FEA_KEGG_2021, cache=TRUE, include=TRUE, echo=FALSE, eval=TRUE, fig.cap="Functional enrichment (KEGG 2021) analysis considering dysregulated genes in the networks of predicted miRNA and protein-coding genes. The heatmaps represent the significance (-log$_{10}$(P-value)) of the most enriched terms (rows) associated to the dysregulated genes found on the seven analyzed cohorts (columns) when considering protein-coding genes with LoF mutations (PCGs::LoF; top-left), protein-coding genes with cis-regulatory mutations (PCG::TFBS; top-right), and miRNAs with cis-regulatory mutations (miRNAs::TFBS; bottom-right). The terms are ordered by their mean ranks across the cohorts.", fig.height = 70, fig.width = 60, fig.align = "center"}

# plot.list <- list()
# load("FEA_heatmaps/GSEA_enrichment_pancancer_heatmap.RData")  ## heatmap.GSEA.list
# plot.list[["PCG::TFBS"]] <- heatmap.GSEA.list[["KEGG_2019_Human"]][["PCG::TFBS"]]
# plot.list[["premiRNA::TFBS"]] <- heatmap.GSEA.list[["KEGG_2019_Human"]][["premiRNA::TFBS"]]
# plot.list[["PCG::LoF"]] <- heatmap.GSEA.list[["KEGG_2019_Human"]][["PCG::LoF"]]

types <- c("PCG::TFBS",
           "premiRNA::TFBS",
           "PCG::LoF")

load("FEA_heatmaps/GSEA_enrichment_pancancer_heatmap_ordered_labels.RData")  ## heatmap.GSEA.axis.list


db <- "KEGG_2021_Human"
font.size <- 55
# co <- "PCG::TFBS"

heatmap.GSEA.list <- list()
for(co in types){
  ti = co
  if(co=="premiRNA::TFBS")
    ti = "miRNA::TFBS"
  fea.df <- heatmap.GSEA.axis.list[[db]][[co]][["df"]]

  cohorts.sorted <- heatmap.GSEA.axis.list[[db]][[co]][["x_axis"]]
  terms.sorted <- heatmap.GSEA.axis.list[[db]][[co]][["y_axis"]]

  heatmap.GSEA.list[[co]] <-
    ggplot(fea.df, aes(x = Cohort, y = Term, fill = Significance)) +
        geom_tile(width = 0.95, height = 0.95) +
        # coord_equal() +
        theme_bw() +
        theme(legend.position = "bottom", legend.title=element_text(size=font.size)) +
        scale_fill_viridis_c(option = "A",direction = -1, breaks = floor(seq(0, max(fea.df$Significance), length.out = 4)), limits = c(0, max(fea.df$Significance))) + # limits = c(0, 75)
        # scale_fill_distiller(palette = "PuRd", direction = +1, breaks = gg.breaks) +
        theme(axis.text.x = element_text(angle = 45, hjust = 1, vjust=1, size=font.size, colour= "black"),
              axis.text.y = element_text(size=font.size, colour = "black"),
              axis.title.y = element_text(size=font.size,colour = "black",vjust=0.12),
              axis.title.x = element_text(size=font.size,colour = "black",vjust=0.12),
              legend.text=element_text(size=font.size),
              plot.title = element_text(size=font.size)) +
        scale_x_discrete(limits = cohorts.sorted) +
        scale_y_discrete(limits = terms.sorted) +
        labs(title = paste0(db, "\n", ti), x = "", y = "") +
        guides(fill = guide_colourbar(barwidth = 17, barheight = 6)) +
        guides(color = guide_legend(title.position = "top",
                                    title.hjust = 0.5,
                                    label.position = "left")) +
    ## Add a border to avoid trunked axis legends
      theme(
        panel.background = element_rect(fill = "white"),
        plot.margin = margin(0.25, 0.25, 0.25, 0.25, "cm"),
        plot.background = element_rect(
          fill = "#fefefe",
          colour = "white",
          size = 1))
}


enriched.terms.gg <- plot_grid(heatmap.GSEA.list[["PCG::LoF"]],
                               heatmap.GSEA.list[["PCG::TFBS"]], NULL,
                               heatmap.GSEA.list[["premiRNA::TFBS"]],
                               ncol = 2, align = "hv", labels = c('', '', ''), label_size = 30,
                               label_colour = "black", hjust = -2, vjust = 3,
                               rel_heights = c(1, 1, 1,1), rel_widths = c(1, 1, 1,1))

## Combine plots
enriched.terms.gg

rm(heatmap.GSEA.list)
rm(enriched.terms.gg)
```



\pagebreak
```{r FEA_WIKIPATH, cache=TRUE, include=TRUE, echo=FALSE, eval=TRUE, fig.cap="Functional enrichment (WikiPathways 2019 human) analysis considering dysregulated genes in the networks of predicted miRNA and protein-coding genes. The heatmaps represent the significance (-log$_{10}$(P-value)) of the most enriched terms (rows) associated to the dysregulated genes found on the seven analyzed cohorts (columns) when considering protein-coding genes with LoF mutations (PCGs::LoF; top-left), protein-coding genes with cis-regulatory mutations (PCG::TFBS; top-right), and miRNAs with cis-regulatory mutations (miRNAs::TFBS; bottom-right). The terms are ordered by their mean ranks across the cohorts.", fig.height = 100, fig.width = 86, fig.align = "center"}

types <- c("PCG::TFBS",
           "premiRNA::TFBS",
           "PCG::LoF")

load("FEA_heatmaps/GSEA_enrichment_pancancer_heatmap_ordered_labels.RData")  ## heatmap.GSEA.axis.list


db <- "WikiPathways_2019_Human"
font.size <- 60
# co <- "PCG::TFBS"
heatmap.GSEA.list <- list()

for(co in types){
  ti = co
  if(co=="premiRNA::TFBS")
    ti = "miRNA::TFBS"
  fea.df <- heatmap.GSEA.axis.list[[db]][[co]][["df"]]

  cohorts.sorted <- heatmap.GSEA.axis.list[[db]][[co]][["x_axis"]]
  terms.sorted <- heatmap.GSEA.axis.list[[db]][[co]][["y_axis"]]

  heatmap.GSEA.list[[co]] <-
    ggplot(fea.df, aes(x = Cohort, y = Term, fill = Significance)) +
        geom_tile(width = 0.95, height = 0.95) +
        # coord_equal() +
        theme_bw() +
        theme(legend.position = "bottom", legend.title=element_text(size=font.size)) +
        scale_fill_viridis_c(option = "A",direction = -1, breaks = floor(seq(0, max(fea.df$Significance), length.out = 4)), limits = c(0, max(fea.df$Significance))) + # limits = c(0, 75)
        # scale_fill_distiller(palette = "PuRd", direction = +1, breaks = gg.breaks) +
        theme(axis.text.x = element_text(angle = 45, hjust = 1, vjust=1, size=font.size, colour= "black"),
              axis.text.y = element_text(size=font.size, colour = "black"),
              axis.title.y = element_text(size=font.size,colour = "black",vjust=0.12),
              axis.title.x = element_text(size=font.size,colour = "black",vjust=0.12),
              legend.text=element_text(size=font.size),
              plot.title = element_text(size=font.size)) +
        scale_x_discrete(limits = cohorts.sorted) +
        scale_y_discrete(limits = terms.sorted) +
        labs(title = paste0(db, "\n", ti), x = "", y = "") +
        guides(fill = guide_colourbar(barwidth = 17, barheight = 6)) +
        guides(color = guide_legend(title.position = "top",
                                    title.hjust = 0.5,
                                    label.position = "left")) +
    ## Add a border to avoid trunked axis legends
      theme(
        panel.background = element_rect(fill = "white"),
        plot.margin = margin(0.25, 0.25, 0.25, 0.25, "cm"),
        plot.background = element_rect(
          fill = "#fefefe",
          colour = "white",
          size = 1))
}


enriched.terms.gg <- plot_grid(heatmap.GSEA.list[["PCG::LoF"]],
                               heatmap.GSEA.list[["PCG::TFBS"]], NULL,
                               heatmap.GSEA.list[["premiRNA::TFBS"]],
                               ncol = 2, align = "hv", labels = c('', '', ''), label_size = 30,
                               label_colour = "black", hjust = -2, vjust = 3,
                               rel_heights = c(1, 1, 1,1), rel_widths = c(1, 1, 1,1))

## Combine plots
enriched.terms.gg

rm(heatmap.GSEA.list)
rm(enriched.terms.gg)
```



\pagebreak
```{r FEA_Panther, cache=TRUE, include=TRUE, echo=FALSE, eval=TRUE, fig.cap="Functional enrichment (Panther 2016) analysis considering dysregulated genes in the networks of predicted miRNA and protein-coding genes. The heatmaps represent the significance (-log$_{10}$(P-value)) of the most enriched terms (rows) associated to the dysregulated genes found on the seven analyzed cohorts (columns) when considering protein-coding genes with LoF mutations (PCGs::LoF; top-left), protein-coding genes with cis-regulatory mutations (PCG::TFBS; top-right), and miRNAs with cis-regulatory mutations (miRNAs::TFBS; bottom-right). The terms are ordered by their mean ranks across the cohorts.", fig.height = 87, fig.width = 77, fig.align = "center"}

types <- c("PCG::TFBS",
           "premiRNA::TFBS",
           "PCG::LoF")

load("FEA_heatmaps/GSEA_enrichment_pancancer_heatmap_ordered_labels.RData")  ## heatmap.GSEA.axis.list


db <- "Panther_2016"
font.size <- 65
# co <- "PCG::TFBS"
heatmap.GSEA.list <- list()

for(co in types){
  ti = co
  if(co=="premiRNA::TFBS")
    ti = "miRNA::TFBS"
  fea.df <- heatmap.GSEA.axis.list[[db]][[co]][["df"]]

  cohorts.sorted <- heatmap.GSEA.axis.list[[db]][[co]][["x_axis"]]
  terms.sorted <- heatmap.GSEA.axis.list[[db]][[co]][["y_axis"]]

  heatmap.GSEA.list[[co]] <-
    ggplot(fea.df, aes(x = Cohort, y = Term, fill = Significance)) +
        geom_tile(width = 0.95, height = 0.95) +
        # coord_equal() +
        theme_bw() +
        theme(legend.position = "bottom", legend.title=element_text(size=font.size)) +
        scale_fill_viridis_c(option = "A",direction = -1, breaks = floor(seq(0, max(fea.df$Significance), length.out = 4)), limits = c(0, max(fea.df$Significance))) + # limits = c(0, 75)
        # scale_fill_distiller(palette = "PuRd", direction = +1, breaks = gg.breaks) +
        theme(axis.text.x = element_text(angle = 45, hjust = 1, vjust=1, size=font.size, colour= "black"),
              axis.text.y = element_text(size=font.size, colour = "black"),
              axis.title.y = element_text(size=font.size,colour = "black",vjust=0.12),
              axis.title.x = element_text(size=font.size,colour = "black",vjust=0.12),
              legend.text=element_text(size=font.size),
              plot.title = element_text(size=font.size)) +
        scale_x_discrete(limits = cohorts.sorted) +
        scale_y_discrete(limits = terms.sorted) +
        labs(title = paste0(db, "\n", ti), x = "", y = "") +
        guides(fill = guide_colourbar(barwidth = 17, barheight = 6)) +
        guides(color = guide_legend(title.position = "top",
                                    title.hjust = 0.5,
                                    label.position = "left")) +
    ## Add a border to avoid trunked axis legends
      theme(
        panel.background = element_rect(fill = "white"),
        plot.margin = margin(0.25, 0.25, 0.25, 0.25, "cm"),
        plot.background = element_rect(
          fill = "#fefefe",
          colour = "white",
          size = 1))
}


enriched.terms.gg <- plot_grid(heatmap.GSEA.list[["PCG::LoF"]],
                               heatmap.GSEA.list[["PCG::TFBS"]], NULL,
                               heatmap.GSEA.list[["premiRNA::TFBS"]],
                               ncol = 2, align = "hv", labels = c('', '', ''), label_size = 30,
                               label_colour = "black", hjust = -2, vjust = 3,
                               rel_heights = c(1, 1, 1,1), rel_widths = c(1, 1, 1,1))

## Combine plots
enriched.terms.gg

rm(heatmap.GSEA.list)
rm(enriched.terms.gg)
```



\pagebreak
\begin{landscape}
```{r miRNAs_predictec_pancancer, cache=TRUE, include=TRUE, echo=FALSE, eval=TRUE, fig.cap="miRNAs predicted by xseq in the seven TCGA cohorts. Cell colors indicate the posterior probability computed over the corresponding cohort. Red stars indicate that the miRNA is annotated as a cancer-associated miRNA in miRCancer. Blue stars indicate that the miRNA was reported as a cancer-associated miRNA in the specific cancer type where it is predicted by xseq, according to miRCancer annotation.", out.width = "650px", out.height = "700px", fig.align = "center"}

knitr::include_graphics("Highlighted_genes_pancancer/xseq_trans_pancancer_heatmap_premiRNA_analysis_mutation_type_TFBS_results_summary_plot.jpeg", dpi = 100)
```
\end{landscape}



\pagebreak
```{r Nb_target_genes_miRNAs, cache=TRUE, include=TRUE, echo=FALSE, eval=TRUE, fig.cap="miRNA-target genes information. A) Distribution of the number of targets considered per miRNA. The red line indicates the median value (1052 targets). The rug at the bottom highlights the predicted miRNAs. B) Number of cohorts (x-axis) where each mature miRNA was predicted versus its number of targets (y-axis).", fig.height = 10, fig.width = 10, fig.align = "center"}

plot.list <- list()
load("Nb_targets_miRNA/Nb_targets_distribution.RData")  ## nb.targets.distr
load("Nb_targets_miRNA/Nb_targets_vs_Nb_cohorts.RData") ## nb.targets.nb.cohorts.violin

plot.list[["Distrib"]] <- nb.targets.distr
plot.list[["Violin"]] <- nb.targets.nb.cohorts.violin


## Add a border to avoid trunked axis legends
for(co in names(plot.list)){

  plot.list[[co]] <-
    plot.list[[co]] +
    theme(
    panel.background = element_rect(fill = "white"),
    plot.margin = margin(0.5, 0.5, 0.5, 0.5, "cm"),
    plot.background = element_rect(
      fill = "#fefefe",
      colour = "white",
      size = 1
    )
  )
}

## Combine plots
suppressMessages(
plot_grid(plot.list[["Distrib"]],
          NULL,
          plot.list[["Violin"]],
          ncol = 1, labels = c('A)', '', 'B)'), label_size = 17, label_colour = "black", hjust = 0, vjust = 3, rel_heights = c(1, 0.15, 1))
)

rm(plot.list)
rm(nb.targets.distr)
rm(nb.targets.nb.cohorts.violin)
```



\pagebreak
```{r Fraction_dysreg_nets, cache=TRUE, include=TRUE, echo=FALSE, eval=TRUE, fig.cap="Fraction of the original gene networks predicted as dysregulated for the predicted genes in A) BRCA-US, B) HNSC-US, C) LIHC-US, D) LUAD-US, E) LUSC-US, F) STAT-US, and G) UCEC-US. X-axis corresponds to the number of dysregulated genes in the filtered network (see Material and methods), Y-axis corresponds to the original size of the gene networks before any filtering (for expression). Color scale corresponds to the fraction of the network that is predicted to be dysregulated in the samples with corresponding mutations of interest.", fig.height = 30, fig.width = 25, fig.align = "center"}

plot.list <- list()

## Load plots
load("Fraction_dysreg_net/Dysregulated_network_fraction_BRCA-US.Rdata") ## dysreg.network.fraction.plot
plot.list[["BRCA"]] <- dysreg.network.fraction.plot + theme_bw()

load("Fraction_dysreg_net/Dysregulated_network_fraction_HNSC-US.Rdata") ## dysreg.network.fraction.plot
plot.list[["HNSC"]] <- dysreg.network.fraction.plot + theme_bw()

load("Fraction_dysreg_net/Dysregulated_network_fraction_LIHC-US.Rdata") ## dysreg.network.fraction.plot
plot.list[["LIHC"]] <- dysreg.network.fraction.plot + theme_bw()

load("Fraction_dysreg_net/Dysregulated_network_fraction_LUAD-US.Rdata") ## dysreg.network.fraction.plot
plot.list[["LUAD"]] <- dysreg.network.fraction.plot + theme_bw()

load("Fraction_dysreg_net/Dysregulated_network_fraction_LUSC-US.Rdata") ## dysreg.network.fraction.plot
plot.list[["LUSC"]] <- dysreg.network.fraction.plot + theme_bw()

load("Fraction_dysreg_net/Dysregulated_network_fraction_STAD-US.Rdata") ## dysreg.network.fraction.plot
plot.list[["STAD"]] <- dysreg.network.fraction.plot + theme_bw()

load("Fraction_dysreg_net/Dysregulated_network_fraction_UCEC-US.Rdata") ## dysreg.network.fraction.plot
plot.list[["UCEC"]] <- dysreg.network.fraction.plot + theme_bw()


## Add a border to avoid trunked axis legends
for(co in names(plot.list)){

  plot.list[[co]] <-
    plot.list[[co]] +
    # scale_colour_gradient2(low = "#feb24c", mid = "#e31a1c", high = "#800026", midpoint = 0.5, limits = c(0,1)) +
    theme(
    panel.background = element_rect(fill = "white"),
    plot.margin = margin(0.75, 0.75, 0.75, 0.75, "cm"),
    plot.background = element_rect(
      fill = "#fefefe",
      colour = "white",
      size = 1
    )
  )
}


suppressWarnings(suppressMessages(
plot_grid(plot.list[["BRCA"]],
          plot.list[["HNSC"]],
          plot.list[["LIHC"]],
          plot.list[["LUAD"]],
          plot.list[["LUSC"]],
          plot.list[["STAD"]],
          plot.list[["UCEC"]],
          ncol = 2, labels = c('A)', 'B)', 'C)', 'D)', 'E)', 'F)', 'G)'), label_size = 35, label_colour = "black", hjust = 0, vjust = 1.15, rel_heights = c(1))
))


rm(plot.list)
rm(dysreg.network.fraction.plot)
```



\pagebreak
\begin{landscape}
```{r FEA_BASIS_1, results='hide', cache=TRUE, include=TRUE, echo=FALSE, eval=TRUE, fig.cap="Functional enrichment analysis considering dysregulated genes in the networks of predicted miRNA and protein-coding genes in the BASIS cohort. The barplots represent the significance (-log$_{10}$(P-value)) of the top-10 most enriched terms (rows) associated to the dysregulated genes when considering protein-coding genes with LoF mutations (PCGs::LoF; A-C), protein-coding genes with cis-regulatory mutations (PCG::TFBS; B-D), and miRNAs with cis-regulatory mutations (miRNAs::TFBS; E). Enriched terms from KEGG 2021 Human (A-B) and WikiPathways (C-E) are provided.", fig.height = 100, fig.width = 180, fig.align = "center"}

load("FEA_heatmaps/Enriched_terms_plots_BASIS.RData")  ## GSEA.enrich.plots

##########
## KEGG ##
##########

## KEGG TFBS PCG
gene.names <- GSEA.enrich.plots[["KEGG_2021_Human"]][["PCG.TFBS"]][["gene_names"]]
basis.pcg.tfbs.kegg <- GSEA.enrich.plots[["KEGG_2021_Human"]][["PCG.TFBS"]][["plot"]]

basis.pcg.tfbs.kegg.plot <-
basis.pcg.tfbs.kegg +
  scale_x_discrete(limits = gene.names) +
  scale_fill_distiller(palette = "Blues", direction = +1) +
  labs(title = " PCG.TFBS - KEGG") +
  theme(text = element_text(size = 90),
        axis.text.x = element_text(hjust=1, size = 100),
        axis.text.y = element_text(hjust=1, size = 100)) +
  guides(fill = guide_colourbar(barwidth = 9, barheight = 30)) +
        theme(
        panel.background = element_rect(fill = "white"),
        plot.margin = margin(0.25, 0.25, 0.25, 0.25, "cm"),
        plot.background = element_rect(
          fill = "#fefefe",
          colour = "white",
          size = 1))


## KEGG LoF
gene.names <- GSEA.enrich.plots[["KEGG_2019_Human"]][["PCG.LoF"]][["gene_names"]]
basis.pcg.lof.kegg <- GSEA.enrich.plots[["KEGG_2019_Human"]][["PCG.LoF"]][["plot"]]

basis.pcg.lof.kegg.plot <-
basis.pcg.lof.kegg +
  scale_x_discrete(limits = gene.names) +
  scale_fill_distiller(palette = "Blues", direction = +1) +
  guides(fill = guide_colourbar(barwidth = 9, barheight = 30)) +
  labs(title = " PCG.LoF - KEGG") +
  theme(text = element_text(size = 100),
          axis.text.x = element_text(hjust=1, size = 100),
          axis.text.y = element_text(hjust=1, size = 100)) +
        theme(
        panel.background = element_rect(fill = "white"),
        plot.margin = margin(0.25, 0.25, 0.25, 0.25, "cm"),
        plot.background = element_rect(
          fill = "#fefefe",
          colour = "white",
          size = 1))


##################
## WikiPathways ##
##################

## WikiPathways TFBS PCG
gene.names <- GSEA.enrich.plots[["WikiPathways_2019_Human"]][["PCG.TFBS"]][["gene_names"]]
basis.pcg.tfbs.wiki <- GSEA.enrich.plots[["WikiPathways_2019_Human"]][["PCG.TFBS"]][["plot"]]

basis.pcg.tfbs.wiki.plot <-
basis.pcg.tfbs.wiki +
  scale_x_discrete(limits = gene.names) +
scale_fill_distiller(palette = "Blues", direction = +1) +
labs(title = "PCG.TFBS - WikiPathways") +
guides(fill = guide_colourbar(barwidth = 9, barheight = 30)) +
theme(text = element_text(size = 100),
          axis.text.x = element_text(hjust=1, size = 100),
          axis.text.y = element_text(hjust=1, size = 100)) +
        theme(
        panel.background = element_rect(fill = "white"),
        plot.margin = margin(0.25, 0.25, 0.25, 0.25, "cm"),
        plot.background = element_rect(
          fill = "#fefefe",
          colour = "white",
          size = 1))


## WikiPathways LoF
gene.names <- GSEA.enrich.plots[["WikiPathways_2019_Human"]][["PCG.LoF"]][["gene_names"]]
basis.pcg.lof.wiki <- GSEA.enrich.plots[["WikiPathways_2019_Human"]][["PCG.LoF"]][["plot"]]

basis.pcg.lof.wiki.plot <-
basis.pcg.lof.wiki +
  scale_x_discrete(limits = gene.names) +
scale_fill_distiller(palette = "Blues", direction = +1) +
guides(fill = guide_colourbar(barwidth = 9, barheight = 30)) +
labs(title = "PCG.LoF - WikiPathways") +
theme(text = element_text(size = 100),
          axis.text.x = element_text(hjust=1, size = 100),
          axis.text.y = element_text(hjust=1, size = 100)) +
        theme(
        panel.background = element_rect(fill = "white"),
        plot.margin = margin(0.25, 0.25, 0.25, 0.25, "cm"),
        plot.background = element_rect(
          fill = "#fefefe",
          colour = "white",
          size = 1))



## WikiPathways TFBS miRNA
gene.names <- GSEA.enrich.plots[["WikiPathways_2019_Human"]][["premiRNA.TFBS"]][["gene_names"]]
basis.premiRNA.tfbs.wiki <- GSEA.enrich.plots[["WikiPathways_2019_Human"]][["premiRNA.TFBS"]][["plot"]]

basis.premiRNA.tfbs.wiki.plot <-
basis.premiRNA.tfbs.wiki +
  scale_x_discrete(limits = gene.names) +
scale_fill_distiller(palette = "Blues", direction = +1) +
guides(fill = guide_colourbar(barwidth = 9, barheight = 30)) +
labs(title = "miRNA.TFBS - WikiPathways") +
theme(text = element_text(size = 100),
          axis.text.x = element_text(hjust=1, size = 100),
          axis.text.y = element_text(hjust=1, size = 100)) #+
#         theme(
#         panel.background = element_rect(fill = "white"),
#         plot.margin = margin(0.25, 0.25, 0.25, 0.25, "cm"),
#         plot.background = element_rect(
#           fill = "#fefefe",
#           colour = "white",
#           size = 1))


enriched.terms.gg <- plot_grid(basis.pcg.lof.kegg.plot,
                               basis.pcg.tfbs.kegg.plot,
                               NULL,
                               NULL, NULL, NULL,
                               basis.pcg.lof.wiki.plot,
                               basis.pcg.tfbs.wiki.plot,
                               basis.premiRNA.tfbs.wiki.plot,
                               ncol = 3, align = "hv", labels = c('A)', 'B)', '', '', '' , '', 'C)', 'D)', 'E)'), label_size = 150,
                               label_colour = "black", hjust = -2, vjust = 3,
                               rel_heights = c(1, 1,1,0,0,0, 1,1,1))

## Combine plots
enriched.terms.gg

rm(GSEA.enrich.plots)
rm(enriched.terms.gg)
```


<!-- ```{r FEA_BASIS_2, cache=TRUE, include=TRUE, echo=FALSE, eval=TRUE, fig.cap="Functional enrichment analysis considering dysregulated genes in the networks of predicted miRNA and protein-coding genes in the BASIS cohort. The heatmaps represent the significance (-log$_{10}$(P-value)) of the top-10 most enriched terms (rows) associated to the dysregulated genes when considering protein-coding genes with LoF mutations (PCGs::LoF; A-C), protein-coding genes with cis-regulatory mutations (PCG::TFBS; B-D), and miRNAs with cis-regulatory mutations (miRNAs::TFBS; E). Enriched terms from GO biological processes (GO-BP; A-B) and Panther 2016 (C-E) are provided.", fig.height = 100, fig.width = 200, fig.align = "center"} -->

<!-- load("FEA_heatmaps/Enriched_terms_plots_BASIS.RData")  ## GSEA.enrich.plots -->

<!-- ########################### -->
<!-- ## GO Biological process ## -->
<!-- ########################### -->

<!-- ## GO-BP TFBS PCG -->
<!-- gene.names <- GSEA.enrich.plots[["GO_Biological_Process_2018"]][["PCG.TFBS"]][["gene_names"]] -->
<!-- basis.pcg.tfbs.go.bp <- GSEA.enrich.plots[["GO_Biological_Process_2018"]][["PCG.TFBS"]][["plot"]] -->

<!-- basis.pcg.tfbs.go.bp.plot <- -->
<!-- basis.pcg.tfbs.go.bp + -->
<!--   scale_x_discrete(limits = gene.names) + -->
<!-- scale_fill_distiller(palette = "Blues", direction = +1) + -->
<!-- guides(fill = guide_colourbar(barwidth = 9, barheight = 30)) + -->
<!-- labs(title = "PCG.TFBS - GO-BP") + -->
<!-- theme(text = element_text(size = 110), -->
<!--           axis.text.x = element_text(hjust=1, size = 110), -->
<!--           axis.text.y = element_text(hjust=1, size = 110)) + -->
<!--         theme( -->
<!--         panel.background = element_rect(fill = "white"), -->
<!--         plot.margin = margin(0.25, 0.25, 0.25, 0.25, "cm"), -->
<!--         plot.background = element_rect( -->
<!--           fill = "#fefefe", -->
<!--           colour = "white", -->
<!--           size = 1)) -->


<!-- ## GO-BP LoF -->
<!-- gene.names <- GSEA.enrich.plots[["GO_Biological_Process_2018"]][["PCG.LoF"]][["gene_names"]] -->
<!-- basis.pcg.lof.go.bp <- GSEA.enrich.plots[["GO_Biological_Process_2018"]][["PCG.LoF"]][["plot"]] -->

<!-- basis.pcg.lof.go.bp.plot <- -->
<!-- basis.pcg.lof.go.bp + -->
<!--   scale_x_discrete(limits = gene.names) + -->
<!-- scale_fill_distiller(palette = "Blues", direction = +1) + -->
<!-- guides(fill = guide_colourbar(barwidth = 9, barheight = 30)) + -->
<!-- labs(title = "PCG.LoF - GO-BP") + -->
<!-- theme(text = element_text(size = 110), -->
<!--           axis.text.x = element_text(hjust=1, size = 110), -->
<!--           axis.text.y = element_text(hjust=1, size = 110)) + -->
<!--         theme( -->
<!--         panel.background = element_rect(fill = "white"), -->
<!--         plot.margin = margin(0.25, 0.25, 0.25, 0.25, "cm"), -->
<!--         plot.background = element_rect( -->
<!--           fill = "#fefefe", -->
<!--           colour = "white", -->
<!--           size = 1)) -->


<!-- ################## -->
<!-- ## Panther 2016 ## -->
<!-- ################## -->

<!-- ## Panther TFBS PCG -->
<!-- gene.names <- GSEA.enrich.plots[["Panther_2016"]][["PCG.TFBS"]][["gene_names"]] -->
<!-- basis.pcg.tfbs.panther <- GSEA.enrich.plots[["Panther_2016"]][["PCG.TFBS"]][["plot"]] -->

<!-- basis.pcg.tfbs.panther.plot <- -->
<!-- basis.pcg.tfbs.panther + -->
<!--   scale_x_discrete(limits = gene.names) + -->
<!-- scale_fill_distiller(palette = "Blues", direction = +1) + -->
<!-- guides(fill = guide_colourbar(barwidth = 9, barheight = 30)) + -->
<!-- labs(title = "PCG.TFBS - Panther") + -->
<!-- theme(text = element_text(size = 110), -->
<!--           axis.text.x = element_text(hjust=1, size = 110), -->
<!--           axis.text.y = element_text(hjust=1, size = 110)) + -->
<!--         theme( -->
<!--         panel.background = element_rect(fill = "white"), -->
<!--         plot.margin = margin(0.25, 0.25, 0.25, 0.25, "cm"), -->
<!--         plot.background = element_rect( -->
<!--           fill = "#fefefe", -->
<!--           colour = "white", -->
<!--           size = 1)) -->


<!-- ## Panther LoF -->
<!-- gene.names <- GSEA.enrich.plots[["Panther_2016"]][["PCG.LoF"]][["gene_names"]] -->
<!-- basis.pcg.lof.panther <- GSEA.enrich.plots[["Panther_2016"]][["PCG.LoF"]][["plot"]] -->

<!-- basis.pcg.lof.panther.plot <- -->
<!-- basis.pcg.lof.panther + -->
<!--   scale_x_discrete(limits = gene.names) + -->
<!-- scale_fill_distiller(palette = "Blues", direction = +1) + -->
<!-- guides(fill = guide_colourbar(barwidth = 9, barheight = 30)) + -->
<!-- labs(title = "PCG.LoF - Panther") + -->
<!-- theme(text = element_text(size = 110), -->
<!--           axis.text.x = element_text(hjust=1, size = 110), -->
<!--           axis.text.y = element_text(hjust=1, size = 110)) + -->
<!--         theme( -->
<!--         panel.background = element_rect(fill = "white"), -->
<!--         plot.margin = margin(0.25, 0.25, 0.25, 0.25, "cm"), -->
<!--         plot.background = element_rect( -->
<!--           fill = "#fefefe", -->
<!--           colour = "white", -->
<!--           size = 1)) -->



<!-- ## Panther TFBS miRNA -->
<!-- gene.names <- GSEA.enrich.plots[["Panther_2016"]][["premiRNA.TFBS"]][["gene_names"]] -->
<!-- basis.premiRNA.tfbs.panther <- GSEA.enrich.plots[["Panther_2016"]][["premiRNA.TFBS"]][["plot"]] -->

<!-- basis.premiRNA.tfbs.panther.plot <- -->
<!-- basis.premiRNA.tfbs.panther + -->
<!--   scale_x_discrete(limits = gene.names) + -->
<!-- scale_fill_distiller(palette = "Blues", direction = +1) + -->
<!-- guides(fill = guide_colourbar(barwidth = 9, barheight = 30)) + -->
<!-- labs(title = "miRNA.TFBS- Panther") + -->
<!-- theme(text = element_text(size = 110), -->
<!--           axis.text.x = element_text(hjust=1, size = 110), -->
<!--           axis.text.y = element_text(hjust=1, size = 110)) + -->
<!--         theme( -->
<!--         panel.background = element_rect(fill = "white"), -->
<!--         plot.margin = margin(0.25, 0.25, 0.25, 0.25, "cm"), -->
<!--         plot.background = element_rect( -->
<!--           fill = "#fefefe", -->
<!--           colour = "white", -->
<!--           size = 1)) -->

<!-- enriched.terms.gg <- plot_grid(basis.pcg.lof.go.bp.plot, -->
<!--                                basis.pcg.tfbs.go.bp.plot, -->
<!--                                NULL, -->

<!--                                NULL,NULL,NULL, -->

<!--                                basis.pcg.lof.panther.plot, -->
<!--                                basis.pcg.tfbs.panther.plot, -->
<!--                                basis.premiRNA.tfbs.panther.plot, -->

<!--                                ncol = 3, align = "hv", labels = c('A)', 'B)', '', '', '', '', "C)", "D)", "E)"), label_size = 150, -->
<!--                                label_colour = "black", hjust = -5, vjust = 1, -->
<!--                                rel_heights = c(1, 1,1, 0,0,0, 1,1,1)) -->

<!-- ## Combine plots -->
<!-- enriched.terms.gg -->

<!-- rm(GSEA.enrich.plots) -->
<!-- rm(enriched.terms.gg) -->
<!-- ``` -->
<!-- \end{landscape} -->


<!-- <!-- \pagebreak --> -->
<!-- <!-- ```{r ER_status_BASIS_vs_TCGA, cache=TRUE, include=TRUE, echo=FALSE, eval=TRUE, fig.cap="ER status repartition between samples in the TCGA BRCA-US and BASIS cohorts.", fig.height = 10, fig.width = 15, fig.align = "center"} --> -->

<!-- <!-- load("ER_status/BASIS_ER_status_tab.RData") ## BASIS.ER.tab --> -->
<!-- <!-- load("ER_status/TCGA_ER_status_tab.RData")  ## TCGA.ER.tab --> -->

<!-- <!-- pie.charts <- list() --> -->

<!-- <!-- ## Custom colors --> -->
<!-- <!-- assigned.colors <- c("#999999", "#E69F00", "#56B4E9") --> -->
<!-- <!-- names(assigned.colors) <- c("Indeterminate", "Positive", "Negative") --> -->

<!-- <!-- for(cohort in c("TCGA", "BASIS")){ --> -->
<!-- <!--   # message("; Generating pie plot for ", cohort) --> -->

<!-- <!--   if(cohort == "TCGA"){ --> -->
<!-- <!--     ER.tab <- TCGA.ER.tab --> -->
<!-- <!--   } else if(cohort == "BASIS"){ --> -->
<!-- <!--     ER.tab <- BASIS.ER.tab --> -->
<!-- <!--   } --> -->

<!-- <!--   ## Ggplot pie chart --> -->
<!-- <!--   pie.chart <- ggplot(ER.tab, aes(x="", y=Freq, fill=ER)) + --> -->
<!-- <!--     geom_bar(width = 1, stat = "identity") + --> -->
<!-- <!--     coord_polar("y", start=0) + --> -->
<!-- <!--     scale_fill_manual(values = assigned.colors[ER.tab$ER], --> -->
<!-- <!--                        breaks= names(assigned.colors[ER.tab$ER])) + --> -->
<!-- <!--     guides(fill=guide_legend(title="ER status")) + --> -->
<!-- <!--     blank_theme + ## From the utilities --> -->
<!-- <!--     theme(axis.text.x=element_blank())+ --> -->
<!-- <!--     ggtitle(paste0(cohort, " BRCA cohort - ", sum(ER.tab$Freq), " selected samples")) + --> -->
<!-- <!--     theme(plot.title = element_text(hjust = 0.5)) + --> -->
<!-- <!--     geom_text(aes(y = Freq/9 + c(0, cumsum(Freq)[-length(Freq)]), --> -->
<!-- <!--                   label = percent(Freq/sum(Freq))), size=5) --> -->


<!-- <!--   pie.charts[[cohort]] <- pie.chart --> -->
<!-- <!-- } --> -->


<!-- <!-- plot_grid(pie.charts[["TCGA"]], pie.charts[["BASIS"]], align = "v") + --> -->
<!-- <!--       theme( --> -->
<!-- <!--     panel.background = element_rect(fill = NA, colour = "#fefefe"), --> -->
<!-- <!--     plot.margin = margin(0.25, 0.25, 0.25, 0.25, "cm"), --> -->
<!-- <!--     plot.background = element_rect( --> -->
<!-- <!--       fill = NA, --> -->
<!-- <!--       colour = NA, --> -->
<!-- <!--       size = 1 --> -->
<!-- <!--     ) --> -->
<!-- <!--   ) --> -->

<!-- <!-- ``` --> -->


<!-- \pagebreak -->
<!-- ```{r Predicted_genes_breast_cancer, cache=TRUE, include=TRUE, echo=FALSE, eval=TRUE, fig.cap="Predicted protein-coding genes in the BASIS cohort using all samples, ER+ samples and ER- samples, and in the TCGA BRCA-US cohort (columns). Predictions were obtained applying the xseq tool on each dataset independently considering protein-coding genes mutated through either LoF (red triangles) or cis-regulatory (TFBS; green triangles) mutations. Genes known as cancer genes (red stars) and TFs (blue stars) are highlighted and hypergeometric tests p-values for enrichment are provided in the legend (Material and methods).", out.width = "650px", out.height="530px", fig.align = "center"} -->

<!-- knitr::include_graphics("BASIS_vs_TCGA/BRCA_predicted_genes.jpeg", dpi = 100) -->
<!-- ``` -->


<!-- \pagebreak -->
<!-- \begin{landscape} -->
<!-- ```{r FEA_breast_cancer_PCG, cache=TRUE, include=TRUE, echo=FALSE, eval=TRUE, fig.cap="Functional enrichment analysis considering dysregulated genes in the networks of predicted protein-coding genes in the breast cancer cohorts (columns). The heatmaps represent the significance (-log$_{10}$(P-value)) of the top-10 most enriched terms (rows) associated to the dysregulated genes for GO biological processes (GO-BP; A), WikiPathways (B), Panther (C), and KEGG (D). Terms (rows) are ordered by their mean rank across all cohorts.", fig.height = 75, fig.width = 115, fig.align = "center"} -->

<!-- ## From: /storage/mathelierarea/processed/jamondra/Projects/dysmir/R_scripts/BASIS/Heatmap_enriched_terms_breast_cancer.R -->


<!-- load("BASIS_vs_TCGA/RData/GSEA_enrichment_pancancer_heatmap_ordered_labels.RData")  ## heatmap.GSEA.axis.list -->

<!-- db1 <- "KEGG_2019_Human" -->
<!-- db2 <- "WikiPathways_2019_Human" -->
<!-- db3 <- "Panther_2016" -->
<!-- db4 <- "GO_Biological_Process_2018" -->
<!-- DBs <- c(db1, db2, db3, db4) -->

<!-- db <- names(heatmap.GSEA.axis.list) -->
<!-- font.size <- 80 -->
<!-- mt <- c("PCG.TFBS") -->
<!-- heatmap.GSEA.list <- list() -->
<!-- for(db in DBs){ -->

<!--   fea.df <- heatmap.GSEA.axis.list[[db]][[mt]][["df"]] -->

<!--   cohorts.sorted <- heatmap.GSEA.axis.list[[db]][[mt]][["x_axis"]] -->
<!--   terms.sorted <- heatmap.GSEA.axis.list[[db]][[mt]][["y_axis"]] -->

<!--   db.title <- NULL -->
<!--   if(db == "KEGG_2019_Human"){ -->
<!--     db.title <- "KEGG" -->
<!--   } -->
<!--   if(db == "WikiPathways_2019_Human"){ -->
<!--     db.title <- "WikiPathways" -->
<!--     } -->
<!--   if(db == "Panther_2016"){ -->
<!--     db.title <- "Panther" -->
<!--   } -->
<!--   if(db == "GO_Biological_Process_2018"){ -->
<!--     db.title <- "GO_BP" -->
<!--   } -->

<!--   heatmap.GSEA.list[[db]] <- -->
<!--     ggplot(fea.df, aes(x = Cohort, y = Term, fill = Significance)) + -->
<!--         geom_tile(width = 0.95, height = 0.95) + -->
<!--         # coord_equal() + -->
<!--         theme_bw() + -->
<!--         theme(legend.position = "bottom", legend.title=element_text(size=font.size)) + -->
<!--         scale_fill_viridis_c(option = "A",direction = -1, breaks = floor(seq(0, max(fea.df$Significance), length.out = 4)), limits = c(0, max(fea.df$Significance))) + # limits = c(0, 75) -->
<!--         # scale_fill_distiller(palette = "PuRd", direction = +1, breaks = gg.breaks) + -->
<!--         theme(axis.text.x = element_text(angle = 45, hjust = 1, vjust=1, size=font.size, colour= "black"), -->
<!--               axis.text.y = element_text(size=font.size, colour = "black"), -->
<!--               axis.title.y = element_text(size=font.size,colour = "black",vjust=0.12), -->
<!--               axis.title.x = element_text(size=font.size,colour = "black",vjust=0.12), -->
<!--               legend.text=element_text(size=font.size), -->
<!--               plot.title = element_text(size=font.size+10)) + -->
<!--         scale_x_discrete(limits = cohorts.sorted) + -->
<!--         scale_y_discrete(limits = terms.sorted) + -->
<!--         labs(title = db.title, x = "", y = "") + -->
<!--         guides(fill = guide_colourbar(barwidth = 30, barheight = 7)) + -->
<!--         guides(color = guide_legend(title.position = "top", -->
<!--                                     title.hjust = 0.5, -->
<!--                                     label.position = "left")) + -->
<!--     ## Add a border to avoid trunked axis legends -->
<!--       theme( -->
<!--         panel.background = element_rect(fill = "white"), -->
<!--         plot.margin = margin(0.5, 0.5, 0.5, 0.5, "cm"), -->
<!--         plot.background = element_rect( -->
<!--           fill = "#fefefe", -->
<!--           colour = "white", -->
<!--           size = 1)) -->
<!-- } -->


<!-- # font.size <- 1 -->
<!-- # null.hm <- ggplot(mtcars, aes(x=cyl)) + -->
<!-- #             geom_bar() + -->
<!-- #             theme(legend.position = "none") -->


<!-- enriched.terms.gg <- plot_grid(heatmap.GSEA.list[[db4]], -->
<!--                                heatmap.GSEA.list[[db2]], -->
<!--                                heatmap.GSEA.list[[db3]], -->
<!--                                heatmap.GSEA.list[[db1]], -->
<!--                                nrow = 2, align = "hv", labels = c('A)', 'B)', 'C)', 'D)'), label_size = 150, -->
<!--                                label_colour = "black", hjust = -7, vjust = 2, -->
<!--                                rel_heights = c(1, 1, 1,1), rel_widths = c(1, 1, 1,1)) -->

<!-- ## Combine plots -->
<!-- enriched.terms.gg -->

<!-- rm(heatmap.GSEA.list) -->
<!-- rm(enriched.terms.gg) -->
<!-- ``` -->


<!-- ```{r FEA_breast_cancer_miRNAs, cache=TRUE, include=TRUE, echo=FALSE, eval=TRUE, fig.cap="Functional enrichment analysis considering dysregulated target genes of predicted miRNAs in the breast cancer cohorts (columns). The heatmaps represent the significance (-log$_{10}$(P-value)) of the top-10 most enriched terms (rows) associated to the dysregulated genes for GO biological processes (GO-BP; A), WikiPathways (B), and Panther (C). Terms (rows) are ordered by their mean rank across all cohorts.", fig.height = 70, fig.width = 115, fig.align = "center"} -->

<!-- ## From: /storage/mathelierarea/processed/jamondra/Projects/dysmir/R_scripts/BASIS/Heatmap_enriched_terms_breast_cancer.R -->


<!-- load("BASIS_vs_TCGA/RData/GSEA_enrichment_pancancer_heatmap_ordered_labels.RData")  ## heatmap.GSEA.axis.list -->

<!-- ## No KEGG for miRNAs -->
<!-- db1 <- "WikiPathways_2019_Human" -->
<!-- db2 <- "Panther_2016" -->
<!-- db3 <- "GO_Biological_Process_2018" -->
<!-- DBs <- c(db1, db2, db3) -->

<!-- db <- names(heatmap.GSEA.axis.list) -->
<!-- font.size <- 80 -->
<!-- mt <- c("premiRNA.TFBS") -->
<!-- heatmap.GSEA.list <- list() -->
<!-- for(db in DBs){ -->

<!--   fea.df <- heatmap.GSEA.axis.list[[db]][[mt]][["df"]] -->

<!--   cohorts.sorted <- heatmap.GSEA.axis.list[[db]][[mt]][["x_axis"]] -->
<!--   terms.sorted <- heatmap.GSEA.axis.list[[db]][[mt]][["y_axis"]] -->

<!--   db.title <- NULL -->
<!--   if(db == "KEGG_2019_Human"){ -->
<!--     db.title <- "KEGG" -->
<!--   } -->
<!--   if(db == "WikiPathways_2019_Human"){ -->
<!--     db.title <- "WikiPathways" -->
<!--     } -->
<!--   if(db == "Panther_2016"){ -->
<!--     db.title <- "Panther" -->
<!--   } -->
<!--   if(db == "GO_Biological_Process_2018"){ -->
<!--     db.title <- "GO_BP" -->
<!--   } -->

<!--   heatmap.GSEA.list[[db]] <- -->
<!--     ggplot(fea.df, aes(x = Cohort, y = Term, fill = Significance)) + -->
<!--         geom_tile(width = 0.95, height = 0.95) + -->
<!--         # coord_equal() + -->
<!--         theme_bw() + -->
<!--         theme(legend.position = "bottom", legend.title=element_text(size=font.size)) + -->
<!--         scale_fill_viridis_c(option = "A",direction = -1, breaks = floor(seq(0, max(fea.df$Significance), length.out = 4)), limits = c(0, max(fea.df$Significance))) + # limits = c(0, 75) -->
<!--         # scale_fill_distiller(palette = "PuRd", direction = +1, breaks = gg.breaks) + -->
<!--         theme(axis.text.x = element_text(angle = 45, hjust = 1, vjust=1, size=font.size, colour= "black"), -->
<!--               axis.text.y = element_text(size=font.size, colour = "black"), -->
<!--               axis.title.y = element_text(size=font.size,colour = "black",vjust=0.12), -->
<!--               axis.title.x = element_text(size=font.size,colour = "black",vjust=0.12), -->
<!--               legend.text=element_text(size=font.size), -->
<!--               plot.title = element_text(size=font.size+10)) + -->
<!--         scale_x_discrete(limits = cohorts.sorted) + -->
<!--         scale_y_discrete(limits = terms.sorted) + -->
<!--         labs(title = db.title, x = "", y = "") + -->
<!--         guides(fill = guide_colourbar(barwidth = 30, barheight = 7)) + -->
<!--         guides(color = guide_legend(title.position = "top", -->
<!--                                     title.hjust = 0.5, -->
<!--                                     label.position = "left")) + -->
<!--     ## Add a border to avoid trunked axis legends -->
<!--       theme( -->
<!--         panel.background = element_rect(fill = "white"), -->
<!--         plot.margin = margin(0.5, 0.5, 0.5, 0.5, "cm"), -->
<!--         plot.background = element_rect( -->
<!--           fill = "#fefefe", -->
<!--           colour = "white", -->
<!--           size = 1)) -->
<!-- } -->

<!-- enriched.terms.gg <- plot_grid(heatmap.GSEA.list[[db3]], -->
<!--                                heatmap.GSEA.list[[db1]], -->
<!--                                heatmap.GSEA.list[[db2]], -->
<!--                                nrow = 2, align = "hv", labels = c('A)', 'B)', 'C)', ''), label_size = 150, -->
<!--                                label_colour = "black", hjust = -7, vjust = 1, -->
<!--                                rel_heights = c(1, 1, 1,1), rel_widths = c(1, 1, 1,1)) -->

<!-- ## Combine plots -->
<!-- enriched.terms.gg -->

<!-- rm(heatmap.GSEA.list) -->
<!-- rm(enriched.terms.gg) -->
<!-- ``` -->
<!-- \end{landscape} -->


<!-- \pagebreak -->
<!-- ```{r Survival_BCSS, cache=TRUE, include=TRUE, echo=FALSE, eval=TRUE, fig.cap="Breast cancer specific survival (BCSS, y-axis) plots for selected miRNAs. Kaplan-Meier survival curves were calculated using the METABRIC cohort. Samples were separated into two groups according to the level of miRNA expression (above/below the median).", fig.height = 30, fig.width = 25, fig.align = "center"} -->

<!-- # setwd("/home/jamondra/Documents/PostDoc/Mathelier_lab/Manuscript/Supp_Material") -->
<!-- load("Survival_plots/BCSS_survival_plots.RData") ## selected.mirnas.bcss.plot -->
<!-- # names(selected.mirnas.bcss.plot) -->

<!-- bcss.examples <- -->
<!-- suppressWarnings( -->
<!-- plot_grid(selected.mirnas.bcss.plot[["hsa-mir-29a-3p"]], -->
<!--           selected.mirnas.bcss.plot[["hsa-mir-1290"]], -->
<!--           selected.mirnas.bcss.plot[["hsa-mir-21-3p"]], -->
<!--           selected.mirnas.bcss.plot[["hsa-mir-181c-3p"]], -->
<!--           selected.mirnas.bcss.plot[["hsa-mir-342-5p"]], -->
<!--           selected.mirnas.bcss.plot[["hsa-mir-145-3p"]], -->
<!--           selected.mirnas.bcss.plot[["hsa-mir-145-5p"]], -->
<!--           selected.mirnas.bcss.plot[["hsa-mir-18a-5p"]], -->
<!--           selected.mirnas.bcss.plot[["hsa-mir-615-3p"]], -->
<!--           selected.mirnas.bcss.plot[["hsa-mir-708-5p"]], -->
<!--           selected.mirnas.bcss.plot[["hsa-mir-29a-5p"]], -->
<!--           selected.mirnas.bcss.plot[["hsa-mir-20a-5p"]], -->
<!--           selected.mirnas.bcss.plot[["hsa-mir-1247-5p"]], -->
<!--           selected.mirnas.bcss.plot[["hsa-mir-17-5p"]], -->
<!--           selected.mirnas.bcss.plot[["hsa-mir-10a-3p"]], -->
<!--           selected.mirnas.bcss.plot[["hsa-mir-92a-1-5p"]], -->
<!--           selected.mirnas.bcss.plot[["hsa-mir-17-3p"]], -->
<!--           selected.mirnas.bcss.plot[["hsa-mir-155-5p"]], -->
<!--           selected.mirnas.bcss.plot[["hsa-mir-339-5p"]], -->
<!--           selected.mirnas.bcss.plot[["hsa-mir-378a-3p"]], -->
<!--           selected.mirnas.bcss.plot[["hsa-mir-21-5p"]], -->
<!--           selected.mirnas.bcss.plot[["hsa-mir-708-3p"]], -->
<!--           selected.mirnas.bcss.plot[["hsa-mir-30b-3p"]], -->
<!--           selected.mirnas.bcss.plot[["hsa-mir-342-3p"]], -->
<!--           nrow = 5) -->
<!-- ) -->

<!-- bcss.examples + -->
<!--       theme( -->
<!--     panel.background = element_rect(fill = NA, colour = "#fefefe"), -->
<!--     plot.margin = margin(0.25, 0.25, 0.25, 0.25, "cm"), -->
<!--     plot.background = element_rect( -->
<!--       fill = NA, -->
<!--       colour = NA, -->
<!--       size = 1 -->
<!--     ) -->
<!--   ) -->

<!-- ``` -->


<!-- \pagebreak -->
<!-- ```{r Survival_OS, cache=TRUE, include=TRUE, echo=FALSE, eval=TRUE, fig.cap="Overall survival (OS, y-axis) plots for selected miRNAs. Kaplan-Meier survival curves were calculated using the METABRIC cohort. Samples were separated into two groups according to the level of miRNA expression (above/below the median).", fig.height = 32, fig.width = 25, fig.align = "center"} -->

<!-- load("Survival_plots/OS_survival_plots.RData") ## selected.mirnas.os.plot -->

<!-- os.examples <- -->
<!-- suppressWarnings( -->
<!-- plot_grid( -->
<!--   selected.mirnas.os.plot[["hsa-mir-20a-5p"]], -->
<!--   selected.mirnas.os.plot[["hsa-mir-29a-3p"]], -->
<!--   selected.mirnas.os.plot[["hsa-mir-145-3p"]], -->
<!--   selected.mirnas.os.plot[["hsa-mir-1290"]], -->
<!--   selected.mirnas.os.plot[["hsa-mir-21-3p"]], -->
<!--   selected.mirnas.os.plot[["hsa-mir-181c-3p"]], -->
<!--   selected.mirnas.os.plot[["hsa-mir-92a-1-5p"]], -->
<!--   selected.mirnas.os.plot[["hsa-mir-145-5p"]], -->
<!--   selected.mirnas.os.plot[["hsa-mir-342-3p"]], -->
<!--   selected.mirnas.os.plot[["hsa-mir-342-5p"]], -->
<!--   selected.mirnas.os.plot[["hsa-mir-155-5p"]], -->
<!--   selected.mirnas.os.plot[["hsa-mir-708-5p"]], -->
<!--   selected.mirnas.os.plot[["hsa-mir-21-5p"]], -->
<!--   selected.mirnas.os.plot[["hsa-mir-10a-3p"]], -->
<!--   selected.mirnas.os.plot[["hsa-mir-30b-3p"]], -->
<!--   selected.mirnas.os.plot[["hsa-mir-615-3p"]], -->
<!--   selected.mirnas.os.plot[["hsa-mir-17-5p"]], -->
<!--   selected.mirnas.os.plot[["hsa-mir-18a-5p"]], -->
<!--   selected.mirnas.os.plot[["hsa-mir-339-5p"]], -->
<!--   selected.mirnas.os.plot[["hsa-mir-17-3p"]], -->
<!--   selected.mirnas.os.plot[["hsa-mir-29a-5p"]], -->
<!--   selected.mirnas.os.plot[["hsa-mir-1247-5p"]], -->
<!--   selected.mirnas.os.plot[["hsa-mir-378a-3p"]], -->
<!--   selected.mirnas.os.plot[["hsa-mir-708-3p"]], -->
<!--           nrow = 6) -->
<!-- ) -->

<!-- suppressWarnings( -->
<!-- os.examples + -->
<!--       theme( -->
<!--     panel.background = element_rect(fill = NA, colour = "#fefefe"), -->
<!--     plot.margin = margin(0.25, 0.25, 0.25, 0.25, "cm"), -->
<!--     plot.background = element_rect( -->
<!--       fill = NA, -->
<!--       colour = NA, -->
<!--       size = 1 -->
<!--     ) -->
<!--   ) -->
<!-- ) -->
<!-- ``` -->












<!-- ###################################################### -->
<!-- ##                                                  ## -->
<!-- ## Old code and code for figures in main manuscript ## -->
<!-- ##                                                  ## -->
<!-- ## Do not run it for the supplementary figures      ## -->
<!-- ##                                                  ## -->
<!-- ###################################################### -->


<!-- \pagebreak -->
<!-- ```{r Nb_associated_TFBSs, cache=TRUE, include=TRUE, echo=FALSE, eval=TRUE, fig.cap="Distributions of the number of associated TFBS to each gene. A) Total (without filtering) number of associated TFBSs. B) Associated TFBSs that are mutated (after filtering).", fig.height = 10, fig.width = 12, fig.align = "center"} -->

<!-- load("Mutation_frequency/Nb_TFBSs_in_genes_distrib.RData") ## nb.tfbs.distrib.plot -->
<!-- load("Mutation_frequency/Nb_mutated_TFBSs_in_genes_distrib.Rdata") ## nb.mutated.tfbs.distrib.plot -->

<!-- suppressMessages( -->
<!-- plot_grid(NULL, -->
<!--           nb.tfbs.distrib.plot, -->
<!--           NULL, -->
<!--           nb.mutated.tfbs.distrib.plot, -->
<!--           ncol = 1, labels = c('', 'A', '', 'B'), label_size = 17, label_colour = "white", hjust = -2, vjust = 3, rel_heights = c(0.1, 1, 0.1, 1)) -->
<!-- ) -->
<!-- ``` -->


<!-- \pagebreak -->
<!-- ```{r Upset_networks_1, cache=TRUE, include=TRUE, echo=FALSE, eval=TRUE, fig.cap="Number of genes in the intersection between the dysregulated networks in A) BRCA-US, B) HNSC-US, C) LIHC-US, and D) LUAD-US.", fig.height = 15, fig.width = 15, fig.align = "center"} -->

<!-- plot.list <- list() -->

<!-- load("Dysreg_net_intersection/Dysreg_networks_intersection_BRCA-US.Rdata")  ## dysreg.genes -->
<!-- plot.list[["BRCA"]] <- dysreg.genes -->

<!-- load("Dysreg_net_intersection/Dysreg_networks_intersection_HNSC-US.Rdata")  ## dysreg.genes -->
<!-- plot.list[["HNSC"]] <- dysreg.genes -->

<!-- load("Dysreg_net_intersection/Dysreg_networks_intersection_LIHC-US.Rdata")  ## dysreg.genes -->
<!-- plot.list[["LIHC"]] <- dysreg.genes -->

<!-- load("Dysreg_net_intersection/Dysreg_networks_intersection_LUAD-US.Rdata")  ## dysreg.genes -->
<!-- plot.list[["LUAD"]] <- dysreg.genes -->


<!-- ## Draw upset plots -->
<!-- up1 <- upset(fromList(plot.list[["BRCA"]]), order.by = "freq", main.bar.color = "#3690c0") -->
<!-- up2 <- upset(fromList(plot.list[["HNSC"]]), order.by = "freq", main.bar.color = "#3690c0") -->
<!-- up3 <- upset(fromList(plot.list[["LIHC"]]), order.by = "freq", main.bar.color = "#3690c0") -->
<!-- up4 <- upset(fromList(plot.list[["LUAD"]]), order.by = "freq", main.bar.color = "#3690c0") -->


<!-- ## Draw titles -->
<!-- theme_georgia <- function(...) { -->
<!--   theme_gray(base_family = "", ...) + -->
<!--     theme(plot.title = element_text(face = "bold")) -->
<!-- } -->

<!-- up1.title <- ggdraw() + -->
<!--   draw_label("Dysregulated genes\nintersection\n\nBRCA-US", -->
<!--              fontfamily = theme_georgia()$text$family, -->
<!--              fontface = theme_georgia()$plot.title$face, x = 0.05, hjust = 0) -->

<!-- up2.title <- ggdraw() + -->
<!--   draw_label("Dysregulated genes\nintersection\n\nHNSC-US", -->
<!--              fontfamily = theme_georgia()$text$family, -->
<!--              fontface = theme_georgia()$plot.title$face, x = 0.05, hjust = 0) -->

<!-- up3.title <- ggdraw() + -->
<!--   draw_label("Dysregulated genes\nintersection\n\nLIHC-US", -->
<!--              fontfamily = theme_georgia()$text$family, -->
<!--              fontface = theme_georgia()$plot.title$face, x = 0.05, hjust = 0) -->

<!-- up4.title <- ggdraw() + -->
<!--   draw_label("Dysregulated genes\nintersection\n\nLUAD-US", -->
<!--              fontfamily = theme_georgia()$text$family, -->
<!--              fontface = theme_georgia()$plot.title$face, x = 0.05, hjust = 0) -->


<!-- ## Print the upset plots (part by part :( ) -->
<!-- up11 <- cowplot::plot_grid(up1.title, up1$Main_bar, up1$Sizes, up1$Matrix, -->
<!--                             nrow=2, align='hv', rel_heights = c(3,1), -->
<!--                            rel_widths = c(2,3)) -->

<!-- up22 <- cowplot::plot_grid(up2.title, up2$Main_bar, up2$Sizes, up2$Matrix, -->
<!--                             nrow=2, align='hv', rel_heights = c(3,1), -->
<!--                            rel_widths = c(2,3)) -->

<!-- up33 <- cowplot::plot_grid(up3.title, up3$Main_bar, up3$Sizes, up3$Matrix, -->
<!--                             nrow=2, align='hv', rel_heights = c(3,1), -->
<!--                            rel_widths = c(2,3)) -->

<!-- up44 <- cowplot::plot_grid(up4.title, up4$Main_bar, up4$Sizes, up4$Matrix, -->
<!--                             nrow=2, align='hv', rel_heights = c(3,1), -->
<!--                            rel_widths = c(2,3)) -->


<!-- ## Combine the upsets in the grid -->
<!-- plot_grid(up11, up22, NULL, NULL, up33, up44, -->
<!--           ncol = 2, -->
<!--           align = "hv", -->
<!--           labels = c('A', 'B', '', '', 'C', 'D'), -->
<!--           label_size = 32, -->
<!--           label_colour = "black", -->
<!--           rel_heights = c(1, 0.35, 1), -->
<!--           vjust = 3 -->
<!--           ) -->


<!-- rm(up1) -->
<!-- rm(up2) -->
<!-- rm(up3) -->
<!-- rm(up4) -->
<!-- rm(up1.title) -->
<!-- rm(up2.title) -->
<!-- rm(up3.title) -->
<!-- rm(up4.title) -->
<!-- rm(up11) -->
<!-- rm(up22) -->
<!-- rm(up33) -->
<!-- rm(up44) -->
<!-- rm(plot.list) -->
<!-- ``` -->



<!-- \pagebreak -->
<!-- ```{r Upset_networks_2, cache=TRUE, include=TRUE, echo=FALSE, eval=TRUE, fig.cap="Number of genes in the intersection between the dysregulated networks in A) LUSC-US, B) STAD-US, and C) UCEC-US.", fig.height = 15, fig.width = 15, fig.align = "center"} -->

<!-- plot.list <- list() -->

<!-- load("Dysreg_net_intersection/Dysreg_networks_intersection_LUSC-US.Rdata")  ## dysreg.genes -->
<!-- plot.list[["LUSC"]] <- dysreg.genes -->

<!-- load("Dysreg_net_intersection/Dysreg_networks_intersection_STAD-US.Rdata")  ## dysreg.genes -->
<!-- plot.list[["STAD"]] <- dysreg.genes -->

<!-- load("Dysreg_net_intersection/Dysreg_networks_intersection_UCEC-US.Rdata")  ## dysreg.genes -->
<!-- plot.list[["UCEC"]] <- dysreg.genes -->


<!-- ## Draw upset plots -->
<!-- up1 <- upset(fromList(plot.list[["LUSC"]]), order.by = "freq", main.bar.color = "#3690c0") -->
<!-- up2 <- upset(fromList(plot.list[["STAD"]]), order.by = "freq", main.bar.color = "#3690c0") -->
<!-- up3 <- upset(fromList(plot.list[["UCEC"]]), order.by = "freq", main.bar.color = "#3690c0") -->


<!-- ## Draw titles -->
<!-- theme_georgia <- function(...) { -->
<!--   theme_gray(base_family = "", ...) + -->
<!--     theme(plot.title = element_text(face = "bold")) -->
<!-- } -->

<!-- up1.title <- ggdraw() + -->
<!--   draw_label("Dysregulated genes\nintersection\n\nLUSC-US", -->
<!--              fontfamily = theme_georgia()$text$family, -->
<!--              fontface = theme_georgia()$plot.title$face, x = 0.05, hjust = 0) -->

<!-- up2.title <- ggdraw() + -->
<!--   draw_label("Dysregulated genes\nintersection\n\nSTAD-US", -->
<!--              fontfamily = theme_georgia()$text$family, -->
<!--              fontface = theme_georgia()$plot.title$face, x = 0.05, hjust = 0) -->

<!-- up3.title <- ggdraw() + -->
<!--   draw_label("Dysregulated genes\nintersection\n\nUCEC-US", -->
<!--              fontfamily = theme_georgia()$text$family, -->
<!--              fontface = theme_georgia()$plot.title$face, x = 0.05, hjust = 0) -->


<!-- ## Print the upset plots (part by part :( ) -->
<!-- up11 <- cowplot::plot_grid(up1.title, up1$Main_bar, up1$Sizes, up1$Matrix, -->
<!--                             nrow=2, align='hv', rel_heights = c(3,1), -->
<!--                            rel_widths = c(2,3)) -->

<!-- up22 <- cowplot::plot_grid(up2.title, up2$Main_bar, up2$Sizes, up2$Matrix, -->
<!--                             nrow=2, align='hv', rel_heights = c(3,1), -->
<!--                            rel_widths = c(2,3)) -->

<!-- up33 <- cowplot::plot_grid(up3.title, up3$Main_bar, up3$Sizes, up3$Matrix, -->
<!--                             nrow=2, align='hv', rel_heights = c(3,1), -->
<!--                            rel_widths = c(2,3)) -->


<!-- ## Combine the upsets in the grid -->
<!-- plot_grid(up11, up22, NULL, NULL, up33, NULL, -->
<!--           ncol = 2, -->
<!--           align = "hv", -->
<!--           labels = c('A', 'B', '', '', 'C', ''), -->
<!--           label_size = 32, -->
<!--           label_colour = "black", -->
<!--           rel_heights = c(1, 0.35, 1), -->
<!--           vjust = 3 -->
<!--           ) -->


<!-- rm(up1) -->
<!-- rm(up2) -->
<!-- rm(up3) -->
<!-- rm(up1.title) -->
<!-- rm(up2.title) -->
<!-- rm(up3.title) -->
<!-- rm(up11) -->
<!-- rm(up22) -->
<!-- rm(up33) -->
<!-- rm(plot.list) -->
<!-- ``` -->



<!-- \pagebreak -->
<!-- ```{r Screening, cache=TRUE, include=TRUE, echo=FALSE, eval=TRUE, fig.cap="Functional screening of highlighted miRNAs in breast cancer cell lines. The breast cancer cell lines KPL4 and JIMT1 were transfected with miRNA mimics (20nM) and assayed for functional effects 72 hours after transfection. a) Apoptosis measured by cleaved PARP(cPARP); b) phosphorylated Akt (pAkt); c) Proliferation measured by Ki67; d) Cell viability. The dashed lines indicate cut-off points that were considered significant (see Methods). Stars (*) denote significant effects. Original data was taken from Leivonen et al.", fig.height = 10, fig.width = 10, fig.align = "center"} -->

<!-- knitr::include_graphics("Screening/Figure_functional_data.jpeg", dpi = 100) -->

<!-- ``` -->


<!-- \pagebreak -->
<!-- ```{r FEA_KEGG_2019_Fig3, cache=TRUE, include=TRUE, echo=FALSE, eval=TRUE, fig.cap="", fig.height = 74, fig.width = 60, fig.align = "center"} -->

<!-- ################################### -->
<!-- ## Figure 3 of dysmiR manuscript ## -->
<!-- ################################### -->

<!-- load("Dysreg_heatmaps_examples/PCG/Dysregulation_heatmaps_BRCA-US.Rdata") ## all.heatmaps.list -->
<!-- all.heatmaps.list.brca <- all.heatmaps.list -->
<!-- ht1 <- all.heatmaps.list.brca[["FUS_TFBS"]] -->

<!-- ################### -->
<!-- ## GSEA heatmaps ## -->
<!-- ################### -->
<!-- types <- c("PCG::TFBS", -->
<!--            "PCG::LoF") -->

<!-- load("FEA_heatmaps/GSEA_enrichment_pancancer_heatmap_ordered_labels.RData")  ## heatmap.GSEA.axis.list -->


<!-- db <- "KEGG_2019_Human" -->
<!-- font.size <- 40 -->
<!-- # co <- "PCG::TFBS" -->
<!-- # co <- "PCG::LoF" -->
<!-- heatmap.GSEA.list <- list() -->
<!-- for(co in types){ -->

<!--   fea.df <- heatmap.GSEA.axis.list[[db]][[co]][["df"]] -->

<!--   cohorts.sorted <- heatmap.GSEA.axis.list[[db]][[co]][["x_axis"]] -->
<!--   terms.sorted <- heatmap.GSEA.axis.list[[db]][[co]][["y_axis"]] -->


<!--   if(co == "PCG::TFBS"){ -->

<!--     terms.sorted[2] <- "Ventricular cardiomyopathy" -->
<!--     terms.sorted[24] <- "Sarcoma-associated herpesvirus" -->
<!--     terms.sorted[30] <- "AGE-RAGE signaling pathway" -->

<!--     terms.sorted[28] <- "T-cell leukemia virus 1 infection" -->
<!--     terms.sorted[17] <- "Atherosclerosis" -->


<!--     fea.df$Term <- gsub(fea.df$Term, pattern = "Arrhythmogenic right ventricular cardiomyopathy \\(ARVC\\)", replacement = "Ventricular cardiomyopathy") -->
<!--     fea.df$Term <- gsub(fea.df$Term, pattern = "Kaposi sarcoma-associated herpesvirus infection" , replacement = "Sarcoma-associated herpesvirus") -->
<!--     fea.df$Term <- gsub(fea.df$Term, pattern = "AGE-RAGE signaling pathway in diabetic complications", replacement = "AGE-RAGE signaling pathway") -->

<!--     fea.df$Term <- gsub(fea.df$Term, pattern = "Human T-cell leukemia virus 1 infection", replacement = "T-cell leukemia virus 1 infection") -->
<!--     fea.df$Term <- gsub(fea.df$Term, pattern = "Fluid shear stress and atherosclerosis", replacement = "Atherosclerosis") -->

<!--   } else if(co == "PCG::LoF"){ -->

<!--     terms.sorted[13] <- "Sarcoma-associated herpesvirus" -->
<!--     terms.sorted[4] <- "Coagulation cascade" -->
<!--     terms.sorted[22] <- "AGE-RAGE signaling pathway" -->
<!--     # terms.sorted[6] <- "Oocyte maturation" -->
<!--     # terms.sorted[1] <- "Phosphatidylinositol signaling" -->

<!--     fea.df$Term <- gsub(fea.df$Term, pattern = "Kaposi sarcoma-associated herpesvirus infection" , replacement = "Sarcoma-associated herpesvirus") -->
<!--     fea.df$Term <- gsub(fea.df$Term, pattern = "Progesterone-mediated oocyte maturation", replacement = "Oocyte maturation") -->
<!--     fea.df$Term <- gsub(fea.df$Term, pattern = "Phosphatidylinositol signaling system", replacement = "Phosphatidylinositol signaling") -->
<!--     fea.df$Term <- gsub(fea.df$Term, pattern = "Complement and coagulation cascades", replacement = "Coagulation cascade") -->
<!--     fea.df$Term <- gsub(fea.df$Term, pattern = "AGE-RAGE signaling pathway in diabetic complications", replacement = "AGE-RAGE signaling pathway") -->
<!--   } -->



<!--   heatmap.GSEA.list[[co]] <- -->
<!--     ggplot(fea.df, aes(x = Cohort, y = Term, fill = Significance)) + -->
<!--         geom_tile(width = 0.95, height = 0.95) + -->
<!--         # coord_equal() + -->
<!--         theme_bw() + -->
<!--         theme(legend.position = "bottom", legend.title=element_text(size=font.size)) + -->
<!--         scale_fill_viridis_c(option = "A",direction = -1, breaks = floor(seq(0, max(fea.df$Significance), length.out = 4)), limits = c(0, max(fea.df$Significance))) + # limits = c(0, 75) -->
<!--         # scale_fill_distiller(palette = "PuRd", direction = +1, breaks = gg.breaks) + -->
<!--         theme(axis.text.x = element_text(angle = 45, hjust = 1, vjust=1, size=font.size, colour= "black"), -->
<!--               axis.text.y = element_text(size=font.size, colour = "black"), -->
<!--               axis.title.y = element_text(size=font.size,colour = "black",vjust=0.12), -->
<!--               axis.title.x = element_text(size=font.size,colour = "black",vjust=0.12), -->
<!--               legend.text=element_text(size=font.size), -->
<!--               plot.title = element_text(size=font.size)) + -->
<!--         scale_x_discrete(limits = cohorts.sorted) + -->
<!--         scale_y_discrete(limits = terms.sorted) + -->
<!--         # labs(title = paste0(db, "\n", co), x = "", y = "") + -->
<!--         labs(title ="", x = "", y = "") + -->
<!--         guides(fill = guide_colourbar(barwidth = 17, barheight = 6)) + -->
<!--         guides(color = guide_legend(title.position = "top", -->
<!--                                     title.hjust = 0.5, -->
<!--                                     label.position = "left")) + -->
<!--     ## Add a border to avoid trunked axis legends -->
<!--       theme( -->
<!--         panel.background = element_rect(fill = "white"), -->
<!--         plot.margin = margin(0.25, 0.25, 0.25, 0.25, "cm"), -->
<!--         plot.background = element_rect( -->
<!--           fill = "#fefefe", -->
<!--           colour = "white", -->
<!--           size = 1)) -->
<!-- } -->


<!-- enriched.terms.gg <- plot_grid(heatmap.GSEA.list[["PCG::LoF"]], -->
<!--                                heatmap.GSEA.list[["PCG::TFBS"]], -->
<!--                                ncol = 2, align = "hv", labels = c('', ''), label_size = 70, -->
<!--                                label_colour = "black", hjust = -2, vjust = 3, -->
<!--                                rel_heights = c(1, 1), rel_widths = c(1,1)) -->

<!-- ## Combine plots -->
<!-- enriched.terms.gg -->
<!-- ##25 x 30 Land -->
<!-- ``` -->


<!-- \pagebreak -->
<!-- ```{r Figure_5_main, cache=TRUE, include=TRUE, echo=FALSE, eval=TRUE, fig.cap="", fig.height = 25, fig.width = 31, fig.align = "center"} -->

<!-- ################################## -->
<!-- ## Figure 5 - dysmiR manuscript ## -->
<!-- ################################## -->

<!-- ## Dysregulation heatmap -->
<!-- load("Dysreg_heatmaps_examples/miRNA/Dysregulation_heatmaps_BRCA-US.Rdata") ## all.heatmaps.list -->
<!-- all.heatmaps.list.brca <- all.heatmaps.list -->
<!-- # names(all.heatmaps.list.brca) -->
<!-- # ht1 <- all.heatmaps.list.brca[["hsa-mir-29a-5p::hsa-mir-29a_TFBS"]] -->
<!-- ht1 <- all.heatmaps.list.brca[["hsa-mir-29a-3p::hsa-mir-29a_TFBS"]] -->
<!-- ## 9.5 X 5.5 -->

<!-- ## GSEA heatmap -->
<!-- load("FEA_heatmaps/GSEA_enrichment_pancancer_heatmap_ordered_labels.RData")  ## heatmap.GSEA.axis.list -->


<!-- co <- "premiRNA::TFBS" -->
<!-- db <- "KEGG_2019_Human" -->
<!-- font.size <- 55 -->
<!-- fea.df <- heatmap.GSEA.axis.list[[db]][[co]][["df"]] -->

<!-- cohorts.sorted <- heatmap.GSEA.axis.list[[db]][[co]][["x_axis"]] -->
<!-- terms.sorted <- heatmap.GSEA.axis.list[[db]][[co]][["y_axis"]] -->

<!-- terms.sorted[7] <- "AGE-RAGE signaling pathway" -->
<!-- terms.sorted[14] <- "HIV 1 infection" -->
<!-- terms.sorted[2] <- "Protein processing in ER" -->

<!-- fea.df$Term <- gsub(fea.df$Term, pattern = "AGE-RAGE signaling pathway in diabetic complications", replacement = "AGE-RAGE signaling pathway") -->
<!-- fea.df$Term <- gsub(fea.df$Term, pattern = "Human immunodeficiency virus 1 infection", replacement = "HIV 1 infection") -->
<!-- fea.df$Term <- gsub(fea.df$Term, pattern = "Protein processing in endoplasmic reticulum", replacement = "Protein processing in ER") -->


<!-- heatmap.GSEA <- -->
<!--     ggplot(fea.df, aes(x = Cohort, y = Term, fill = Significance)) + -->
<!--         geom_tile(width = 0.95, height = 0.95) + -->
<!--         # coord_equal() + -->
<!--         theme_bw() + -->
<!--         theme(legend.position = "bottom", legend.title=element_text(size=font.size)) + -->
<!--         scale_fill_viridis_c(option = "A",direction = -1, breaks = floor(seq(0, max(fea.df$Significance), length.out = 4)), limits = c(0, max(fea.df$Significance))) + # limits = c(0, 75) -->
<!--         # scale_fill_distiller(palette = "PuRd", direction = +1, breaks = gg.breaks) + -->
<!--         theme(axis.text.x = element_text(angle = 45, hjust = 1, vjust=1, size=font.size, colour= "black"), -->
<!--               axis.text.y = element_text(size=font.size, colour = "black"), -->
<!--               axis.title.y = element_text(size=font.size,colour = "black",vjust=0.12), -->
<!--               axis.title.x = element_text(size=font.size,colour = "black",vjust=0.12), -->
<!--               legend.text=element_text(size=font.size-10, margin = margin(t = 10)), -->
<!--               legend.spacing.x = unit(1, 'cm'), -->
<!--               plot.title = element_text(size=font.size)) + -->
<!--         scale_x_discrete(limits = cohorts.sorted) + -->
<!--         scale_y_discrete(limits = terms.sorted) + -->
<!--         labs(title = paste0(db, "\n", co), x = "", y = "") + -->
<!--         guides(fill = guide_colourbar(barwidth = 17, barheight = 5)) + -->
<!--         guides(colour = guide_legend(title.position = "right", -->
<!--                                     title = "", -->
<!--                                     title.hjust = 0.5, -->
<!--                                     label.position = "bottom")) + -->
<!--         theme(legend.title=element_blank()) -->
<!-- # heatmap.GSEA -->
<!-- ## 40 x 23 -->
<!-- ``` -->


<!-- \pagebreak -->
<!-- ```{r Aggregated_networks, cache=TRUE, include=TRUE, echo=FALSE, eval=TRUE, fig.cap="", fig.height = 18, fig.width = 15, fig.align = "center"} -->

<!-- # setwd("../..") -->
<!-- load("Indiv_nets/Nb_subgraphs_individual_samples_all_cohorts.Rdata") -->
<!-- subgraphs <- xx -->

<!-- load("Indiv_nets/UCEC-US/Aggregated_network_UCEC-US.Rdata") -->
<!-- agg.net.ucec <- agg.net + add_space + theme(plot.title = element_text(size=22)) -->


<!-- agg.nets.fig <- -->
<!-- plot_grid(subgraphs, -->
<!--           agg.net.ucec, -->
<!--           nrow = 1, labels = c('A)', 'B)'), -->
<!--           label_size = 45, vjust = 2, hjust = 0, rel_widths = c(0.666, 1)) -->
<!-- agg.nets.fig -->
<!-- ## 30 * 17 P -->
<!-- ## 27 * 11 P -->

<!-- ``` -->




<!-- \pagebreak -->
<!-- ```{r Dysregulation_HeatMaps_miRNAs, cache=TRUE, include=TRUE, echo=FALSE, eval=TRUE, fig.cap="Examples of dysregulated networks for predicted miRNA driver genes through cis-regulatory mutations: hsa-mir-17-3p in BRCA-US (left), hsa-mir-155-5p in LIHC-US (center), and hsa-mir-21-3p in UCEC-US (right). Each heatmap column corresponds to a sample where TFBSs associated to the predicted drivers are found with a somatic mutation. Each row corresponds to dysregulated genes in the network of the predicted driver. The color scale represents the gene regulatory status (red: up-regulation; blue: down-regulation). The horizontal bars on top of the heatmaps show the patient-specific driver posterior probability; the darker the higher the probability. Note patient stratification based on this probability.", fig.height = 29, fig.width = 25, fig.align = "center"} -->

<!-- ## Load the heatmap objects -->

<!-- load("Dysreg_heatmaps_examples/miRNA/Dysregulation_heatmaps_BRCA-US.Rdata") ## all.heatmaps.list -->
<!-- all.heatmaps.list.brca <- all.heatmaps.list -->
<!-- # names(all.heatmaps.list.brca) -->

<!-- load("Dysreg_heatmaps_examples/miRNA/Dysregulation_heatmaps_LUSC-US.Rdata") ## all.heatmaps.list -->
<!-- all.heatmaps.list.lusc <- all.heatmaps.list -->
<!-- # names(all.heatmaps.list.lusc) -->

<!-- load("Dysreg_heatmaps_examples/miRNA/Dysregulation_heatmaps_UCEC-US.Rdata") ## all.heatmaps.list -->
<!-- all.heatmaps.list.ucec <- all.heatmaps.list -->
<!-- # names(all.heatmaps.list.brca) -->
<!-- # names(all.heatmaps.list.lihc) -->
<!-- # names(all.heatmaps.list.ucec) -->

<!-- ht1 <- all.heatmaps.list.brca[["hsa-mir-17-5p::hsa-mir-17_TFBS"]] -->
<!-- ht2 <- all.heatmaps.list.lusc[["hsa-mir-205-5p::hsa-mir-205_TFBS"]] -->
<!-- ht3 <- all.heatmaps.list.ucec[["hsa-mir-21-3p::hsa-mir-21_TFBS"]] -->

<!-- ht11 <- as.ggplot(grid.grabExpr(draw(ht1))) + add_space -->
<!-- ht22 <- as.ggplot(grid.grabExpr(draw(ht2))) + add_space -->
<!-- ht33 <- as.ggplot(grid.grabExpr(draw(ht3))) + add_space -->

<!-- plot_grid(ht11, ht22, ht33, -->
<!--           nrow =  1) -->


<!-- rm(ht11) -->
<!-- rm(ht22) -->
<!-- rm(ht33) -->
<!-- rm(ht1) -->
<!-- rm(ht2) -->
<!-- rm(ht3) -->
<!-- rm(all.heatmaps.list) -->
<!-- rm(all.heatmaps.list.brca) -->
<!-- rm(all.heatmaps.list.lusc) -->
<!-- rm(all.heatmaps.list.ucec) -->
<!-- ``` -->


<!-- \pagebreak -->
<!-- ```{r Dysregulation_HeatMaps_hsa_mir_92a, cache=TRUE, include=TRUE, echo=FALSE, eval=TRUE, fig.cap="Examples of dysregulated networks for predicted miRNA driver genes through cis-regulatory mutations: hsa-mir-92a-1-5p in BRCA-US (left), LIHC-US (center), and UCEC-US (right). Each heatmap column corresponds to a sample where TFBSs associated to the predicted drivers are found with a somatic mutation. Each row corresponds to dysregulated genes in the network of the predicted driver. The color scale represents the gene regulatory status (red: up-regulation; blue: down-regulation). The horizontal bars on top of the heatmaps show the patient-specific driver posterior probability; the darker the higher the probability. Note patient stratification based on this probability.", fig.height = 29, fig.width = 25, fig.align = "center"} -->

<!-- ## Load the heatmap objects -->

<!-- load("Dysreg_heatmaps_examples/miRNA/Dysregulation_heatmaps_BRCA-US.Rdata") ## all.heatmaps.list -->
<!-- all.heatmaps.list.brca <- all.heatmaps.list -->

<!-- load("Dysreg_heatmaps_examples/miRNA/Dysregulation_heatmaps_UCEC-US.Rdata") ## all.heatmaps.list -->
<!-- all.heatmaps.list.ucec <- all.heatmaps.list -->

<!-- load("Dysreg_heatmaps_examples/miRNA/Dysregulation_heatmaps_LIHC-US.Rdata") ## all.heatmaps.list -->
<!-- all.heatmaps.list.lihc <- all.heatmaps.list -->
<!-- # names(all.heatmaps.list.brca) -->
<!-- # names(all.heatmaps.list.lihc) -->
<!-- # names(all.heatmaps.list.ucec) -->

<!-- ht1 <- all.heatmaps.list.brca[["hsa-mir-92a-1-5p::hsa-mir-92a-1_TFBS"]] -->
<!-- ht2 <- all.heatmaps.list.lihc[["hsa-mir-92a-1-5p::hsa-mir-92a-1_TFBS"]] -->
<!-- ht3 <- all.heatmaps.list.ucec[["hsa-mir-92a-1-5p::hsa-mir-92a-1_TFBS"]] -->

<!-- ht11 <- as.ggplot(grid.grabExpr(draw(ht1))) + add_space -->
<!-- ht22 <- as.ggplot(grid.grabExpr(draw(ht2))) + add_space -->
<!-- ht33 <- as.ggplot(grid.grabExpr(draw(ht3))) + add_space -->

<!-- plot_grid(ht11, ht22, ht33, -->
<!--           nrow =  1) -->

<!-- rm(ht11) -->
<!-- rm(ht22) -->
<!-- rm(ht33) -->
<!-- rm(ht1) -->
<!-- rm(ht2) -->
<!-- rm(ht3) -->
<!-- rm(all.heatmaps.list) -->
<!-- rm(all.heatmaps.list.brca) -->
<!-- rm(all.heatmaps.list.lihc) -->
<!-- rm(all.heatmaps.list.ucec) -->
<!-- ``` -->



<!-- \pagebreak -->
<!-- ```{r Selected_genes_BASIS, cache=TRUE, include=TRUE, echo=FALSE, eval=TRUE, fig.cap="Ranking (x-axis) of the posterior probability (y-axis) of likely association between cis-regulatory mutations and network dysregulation (i.e. driver potential) computed by xseq for miRNA (A) and protein-coding (PCG, B) genes in the BASIS cohort. The dashed lines correspond to a threshold of 0.8 on the posterior probability. Names of the top-10 genes are provided for each panel.", fig.height = 25, fig.width = 31, fig.align = "center"} -->

<!-- load("xseq_ranked_genes/BASIS/All/Ranked_PCGs_BASIS_mutation_type_TFBS.RData") ## PCG.ranks.plot -->
<!-- load("xseq_ranked_genes/BASIS/All/Ranked_miRNAs_BASIS_mutation_type_TFBS.RData")   ## mirna.ranks.plot -->


<!-- ## Add a border to avoid trunked axis legends -->
<!-- mirna.ranks.plot <- -->
<!--   mirna.ranks.plot + -->
<!--   theme( -->
<!--     panel.background = element_rect(fill = "white"), -->
<!--     plot.margin = margin(0.5, 0.5, 0.5, 0.5, "cm"), -->
<!--     plot.background = element_rect( -->
<!--       fill = "#fefefe", -->
<!--       colour = "white", -->
<!--       size = 1 -->
<!--     ) -->
<!--   ) + -->
<!--   theme(axis.text.x = element_text(size=30), -->
<!--         axis.text.y = element_text(size=30), -->
<!--         axis.title.x = element_text(size=30), -->
<!--         axis.title.y = element_text(size=30), -->
<!--         legend.text=element_text(size = 25)) + -->
<!--   scale_colour_gradient(low = "#7fcdbb", high = "#253494") + -->
<!--   ylim(c(0,1.1)) + -->
<!--   guides(colour = guide_colourbar(barwidth = 25, barheight = 3)) + -->
<!--   theme(legend.position = "bottom", legend.title=element_text(size=30)) -->


<!-- ## Add a border to avoid trunked axis legends -->
<!-- PCG.ranks.plot <- -->
<!--   PCG.ranks.plot + -->
<!--   theme( -->
<!--     panel.background = element_rect(fill = "white"), -->
<!--     plot.margin = margin(0.5, 0.5, 0.5, 0.5, "cm"), -->
<!--     plot.background = element_rect( -->
<!--       fill = "#fefefe", -->
<!--       colour = "white", -->
<!--       size = 1 -->
<!--     ) -->
<!--   ) + -->
<!--   theme(axis.text.x = element_text(size=30), -->
<!--         axis.text.y = element_text(size=30), -->
<!--         axis.title.x = element_text(size=30), -->
<!--         axis.title.y = element_text(size=30), -->
<!--         legend.text=element_text(size = 25)) + -->
<!--   scale_colour_gradient(low = "#7fcdbb", high = "#253494") + -->
<!--   ylim(c(0,1.1)) + -->
<!--   guides(colour = guide_colourbar(barwidth = 25, barheight = 3)) + -->
<!--   theme(legend.position = "bottom", legend.title=element_text(size=30)) -->

<!-- suppressMessages( -->
<!-- plot_grid(mirna.ranks.plot, -->
<!--           PCG.ranks.plot, -->
<!--           ncol = 2, labels = c('A)', 'B)'), label_size = 50, label_colour = "black", hjust = -0.5, vjust = 3, rel_heights = c(1, 1)) -->
<!-- ) -->
<!-- ``` -->