---
title: Cis-regulatory mutations associate with transcriptional and post-transcriptional deregulation of the gene regulatory program in cancers
author:
- Jaime A. Castro-Mondragon,
- Miriam Ragle Aure,
- Ole Christian Lingjærde,
- Anita Langerød,
- John W. M. Martens,
- Anne-Lise Børresen-Dale,
- Vessela Kristensen,
- and Anthony Mathelier
subtitle: 'Supplemental Material'
date: 'Last update: `r Sys.Date()`'
geometry: margin=1.5cm
output:
  pdf_document
bibliography: References.bib
biblio-style: apalike
link-citations: yes
documentclass: article
#classoption: openany
header-includes: 
  \usepackage{float} \floatplacement{figure}{H} 
  \newcommand{\beginsupplement}{\setcounter{table}{0}
  \renewcommand{\thetable}{S\arabic{table}} \setcounter{figure}{0}
  \renewcommand{\thefigure}{S\arabic{figure}}}
  \usepackage{fullpage}
  \usepackage{pdflscape}
  \usepackage{inputenc}
---

\beginsupplement


```{r setup, include=FALSE, cache=TRUE, echo=FALSE, eval=TRUE}
knitr::opts_chunk$set(echo = TRUE)
suppressPackageStartupMessages(library(bookdown))
suppressPackageStartupMessages(library(Cairo))
suppressPackageStartupMessages(library(ComplexHeatmap))
suppressPackageStartupMessages(library(cowplot))
suppressPackageStartupMessages(library(data.table))
suppressPackageStartupMessages(library(dplyr))
suppressPackageStartupMessages(library(emojifont))
suppressPackageStartupMessages(library(GGally))
suppressPackageStartupMessages(library(ggExtra))
suppressPackageStartupMessages(library(ggnetwork))
suppressPackageStartupMessages(library(ggplot2))
suppressPackageStartupMessages(library(ggplotify))
suppressPackageStartupMessages(library(ggpubr))
suppressPackageStartupMessages(library(ggrepel))
suppressPackageStartupMessages(library(ggsignif))
# suppressPackageStartupMessages(library(leaflet))
suppressPackageStartupMessages(library(scales))
suppressPackageStartupMessages(library(tidyr))
suppressPackageStartupMessages(library(UpSetR))
suppressPackageStartupMessages(library(xseq))

pdf.options(encoding = 'ISOLatin2')
options(device = "cairo_pdf")

cohorts <- c("BRCA", "HNSC", "LIHC", "LUAD", "LUSC", "STAD", "UCEC")


## Custom
blank_theme <- theme_minimal() +
  theme(
    axis.title.x = element_blank(),
    axis.title.y = element_blank(),
    panel.border = element_blank(),
    panel.grid=element_blank(),
    axis.ticks = element_blank(),
    plot.title=element_text(size=14, face="bold")
  )

add_space <-   theme(
    panel.background = element_rect(fill = "white", colour = "white"),
    plot.margin = margin(0.5, 0.5, 0.5, 0.5, "cm"),
    plot.background = element_rect(
      fill = "#FFFEFD",
      colour = "#FFFEFD",
      size = 0.001
    )
  )


add_space_gg <-   theme(
    panel.background = element_rect(fill = "white"),
    plot.margin = margin(0.5, 0.5, 0.5, 0.5, "cm"),
    plot.background = element_rect(
      fill = "#FFFEFD",
      colour = "white",
      size = 1
    ))


agg.net.fun <- function(df           = NULL,
                        project.name = NULL,
                        clusters     = NULL) {

  ggplot(df, aes(x = x, y = y, xend = xend, yend = yend, color = Nb_samples, fill = TF_gene), arrow.gap = 0) +
        geom_edges(size = 1, colour = "#6baed6", ) +
        geom_nodes(size = 8) +
        geom_nodetext(aes(label = Nb_samples), fontface = "bold", color = "white", size = 5) +
        scale_color_distiller(palette = "PuBu", direction = +1, limits = c(1 ,max(df$Nb_samples)), name = "Samples    ") + 
        theme_blank() +
        scale_fill_manual(values = c("#fdbb84", "white"),
                          breaks = c("TF", "non_TF")) + ##
        labs(title = paste("Aggregated network ", project.name, " - Genes: ", sum(clusters$csize), " - Subgraphs: ", clusters$no)) +
        guides(fill = "none") +
        geom_nodelabel_repel(aes(label = vertex.names), fontface = "plain", box.padding = unit(1, "lines"), color = "black", force = 5, size = 6, point.padding = unit(1.5, "lines")) +
        theme(plot.title = element_text(hjust = 0.5, size=12),
              legend.text=element_text(size=10),
              legend.title=element_text(size=10),
              legend.position = "bottom",
              legend.box = "vertical") +
    guides(colour = "none")
        # guides(colour = guide_colourbar(barwidth = 7, barheight = 2))
    
}


xseq.th.pcg <- list('BRCA-US'   = 0.5,
                'HNSC-US'   = 0.5,
                'LIHC-US'   = 0.5,
                'LUAD-US'   = 0.8936041,
                'LUSC-US'   = 0.8776286,
                'STAD-US'   = 0.9154082,
                'UCEC-US'   = 0.9717390,
                "BASIS_All" = 0.5,
                "BASIS_ER+" = 0.5,
                "BASIS_ER-" = 0.9837910)


xseq.th.mirna <- list('BRCA-US'   = 0.5,
                'HNSC-US'   = 0.9999984,
                'LIHC-US'   = 0.9990923,
                'LUAD-US'   = 0.9999987,
                'LUSC-US'   = 0.9999835,
                'STAD-US'   = 0.9999981,
                'UCEC-US'   = 0.9999773,
                "BASIS_All" = 0.5,
                "BASIS_ER+" = 0.5,
                "BASIS_ER-" = 0.9837910)

load("Nb_TFBSs_per_gene/TFBS_associated_with_GeneHancer_target_genes_PCG_miRNA_hg19.RData")
load("Nb_TFBSs_per_gene/Xseq_supp_material_tables.Rdata")


xseq.net.to.df <- function(net) {
  
  gene.names <- names(net)
  
  net.df <- lapply(seq_along(gene.names), function(i){
    
    df <- data.frame(net[[i]])
    
    df <- df %>% 
      mutate(Gene = gene.names[i], Target = rownames(df))
    df <- df[,c(2,3,1)]
    colnames(df) <- c("Gene", "Target", "Weight")
    
    df
  })
  
  net.df <- do.call(rbind, net.df)
  
  return(net.df)
}


####################################
## Load miRNA <-> pre-miRNA table ##
####################################
dict.load <- load("miR_premiR_dict/hsa_mir_to_premir.RData")
mirna.premirna.dict <- get(dict.load)
rm(dict.load)
mirna.premirna.dict$miR    <- tolower(mirna.premirna.dict$miR)
mirna.premirna.dict$premiR <- tolower(mirna.premirna.dict$premiR)
## Add a column with a unique Id for the mature miRNAs
mirna.premirna.dict <- mirna.premirna.dict %>% 
                        mutate(mature_ID = paste(miR, premiR, sep = "::"))
## Get the names for those miRNAs encoded by two or more precursors
mirnas.two.precursors <- mirna.premirna.dict$miR[duplicated(mirna.premirna.dict$miR)]


######################
## Xseq ppi network ##
######################
load("Nb_connections_per_gene/xseq_PPI_network.Rdata") ## network.to.export
xseq.net.ppi <- network.to.export
xseq.net.ppi <- FilterNetwork(xseq.net.ppi, max.num.conn = 1000, min.conn.strength = 0.8)
xseq.net.ppi <- xseq.net.to.df(xseq.net.ppi)



########################
## Xseq miRNA network ##
########################
load("Nb_connections_per_gene/targetScan_clean_miRNA-mRNA_network.Rdata") ## network.to.export
xseq.net.mirna <- network.to.export
xseq.net.mirna <- FilterNetwork(xseq.net.mirna, max.num.conn = 100, min.conn.strength = 0.8)
xseq.net.mirna <- xseq.net.to.df(xseq.net.mirna)





# Laptop -> Biotin
# rsync -rptuvl /home/jamondra/Documents/PostDoc/Mathelier_lab/Manuscript/Supp_Material jamondra@biotin2.hpc.uio.no:/storage/mathelierarea/processed/jamondra/Projects/dysmir/Manuscript

# Biotin -> Laptop
# rsync -rptuvl jamondra@biotin2.hpc.uio.no:/storage/mathelierarea/processed/jamondra/Projects/dysmir/Manuscript/Supp_Material /home/jamondra/Documents/PostDoc/Mathelier_lab/Manuscript
```



\pagebreak
```{r Number_of_Mutations, include=TRUE, cache=TRUE, echo=FALSE, eval=TRUE, fig.cap="A) Number of mutations per sample per cohort. B) Number of cis-regulatory mutations (i.e. within TFBSs) per sample per cohort.", fig.height = 8, fig.width = 6, fig.align = "center"}

load("Mutation_frequency/Nb_mutated_nt_within_TFBSs_across_cohorts.RData")
load("Mutation_frequency/Nb_mutations_per_cohort_dotplot.RData")

wgs.mut.summary.plot <-
  wgs.mut.summary.plot +
  theme(
    panel.background = element_rect(fill = "white"),
    plot.margin = margin(0.5, 0.5, 0.5, 0.5, "cm"),
    plot.background = element_rect(
      fill = "#fefefe",
      colour = "white",
      size = 1
    )
  ) +
  labs(title = "Number of mutations")


mutated.tfbs.cohorts <-
  mutated.tfbs.cohorts +
  theme(
    panel.background = element_rect(fill = "white"),
    plot.margin = margin(0.5, 0.5, 0.5, 0.5, "cm"),
    plot.background = element_rect(
      fill = "#fefefe",
      colour = "white",
      size = 1
    )
  ) +
  labs(title = "Number of mutations in TFBSs")

suppressWarnings(
plot_grid(wgs.mut.summary.plot, mutated.tfbs.cohorts, ncol = 1, labels = c('A)', 'B)'), label_size = 17, align = "v", hjust = -0, vjust = 2.9)
)
```





\pagebreak
```{r Mutation_fraction_in_TFBSs, include=TRUE, echo=FALSE, eval=TRUE, fig.cap="Fraction of somatic mutations per sample per cohort categorized as cis-regulatory mutations (i.e. within TFBSs).", fig.height = 5, fig.width = 7, fig.align = "center"}

library(ggplot2)
load("Mutation_frequency/Fraction_of_somatic_mutations_in_TFBSs.RData")

Fraction.mut.in.tfbs.gg <- Fraction.mut.in.tfbs.gg +
                            theme_classic() +
                            theme(
                              panel.background = element_rect(fill = "white"),
                              # plot.margin = margin(0.5, 0.5, 0.5, 0.5, "cm"),
                              plot.background = element_rect(
                                fill = "#fefefe",
                                colour = "white",
                                size = 1
                              )
                            ) +
                            theme(legend.position = "none") 

suppressWarnings(print(Fraction.mut.in.tfbs.gg))

```





\pagebreak
```{r Mutation_rates_paired_1, cache=TRUE, include=TRUE, echo=FALSE, eval=TRUE, fig.cap="Paired comparisons (wilcoxon tests) of mutation rates in A) BRCA samples (n = 92), B) LIHC (n = 50), C) UCEC (n = 48), and D) HNSC (n = 43) at TFBS, exonic, flanking, and random regions.", fig.height = 19, fig.width = 15, fig.align = "center"}

load("Mutation_rate/Paired_Comparison_Mutation_rates_BRCA-US.RData")
mut.rate.distrib.comp.gg.brca <- mut.rate.distrib.comp.gg + add_space

load("Mutation_rate/Paired_Comparison_Mutation_rates_LIHC-US.RData")
mut.rate.distrib.comp.gg.lihc <- mut.rate.distrib.comp.gg + add_space

load("Mutation_rate/Paired_Comparison_Mutation_rates_UCEC-US.RData")
mut.rate.distrib.comp.gg.ucec <- mut.rate.distrib.comp.gg + add_space

load("Mutation_rate/Paired_Comparison_Mutation_rates_HNSC-US.RData")
mut.rate.distrib.comp.gg.hnsc <- mut.rate.distrib.comp.gg + add_space

suppressMessages(suppressWarnings(plot_grid(mut.rate.distrib.comp.gg.brca, 
                                            mut.rate.distrib.comp.gg.lihc,
                                            mut.rate.distrib.comp.gg.ucec,
                                            mut.rate.distrib.comp.gg.hnsc,
                                            ncol = 1,
                                            nrow = 4,
                                            labels = c('A)', 'B)', 'C)', 'D)'),
                                            label_size = 17,
                                            align = "hv",
                                            rel_heights = c(0.8),
                                            vjust = 1)))
```





\pagebreak
```{r Mutation_rates_paired_2, cache=TRUE, include=TRUE, echo=FALSE, eval=TRUE, fig.cap="Paired comparisons (wilcoxon tests) of mutation rates in A) STAD samples (n = 35), B) LUAD (n = 37), and C) LUSC (n = 42) at TFBS, exonic, flanking, and random regions.", fig.height = 19, fig.width = 15, fig.align = "center"}

load("Mutation_rate/Paired_Comparison_Mutation_rates_STAD-US.RData")
mut.rate.distrib.comp.gg.stad <- mut.rate.distrib.comp.gg + add_space

load("Mutation_rate/Paired_Comparison_Mutation_rates_LUAD-US.RData")
mut.rate.distrib.comp.gg.luad <- mut.rate.distrib.comp.gg + add_space

load("Mutation_rate/Paired_Comparison_Mutation_rates_LUSC-US.RData")
mut.rate.distrib.comp.gg.lusc <- mut.rate.distrib.comp.gg + add_space

suppressMessages(suppressWarnings(plot_grid(mut.rate.distrib.comp.gg.stad, 
                                            mut.rate.distrib.comp.gg.luad,
                                            mut.rate.distrib.comp.gg.lusc,
                                            ncol = 1,
                                            nrow = 3,
                                            labels = c('A)', 'B)', 'C)'),
                                            label_size = 17,
                                            align = "hv",
                                            rel_heights = c(0.9),
                                            vjust = 1)))
```





\pagebreak
```{r TFBS_vs_Exonic_mutation_rate_BRCA, cache=TRUE, include=TRUE, echo=FALSE, eval=TRUE, fig.cap="TFBS versus exonic mutation rates in TCGA cohorts. Both axis are -log$_{10}$ converted. The diagonal line indicates identical mutation rates in TFBS and exonic regions. The blue line represents a linear regression model best fitting the points with a 95\\% confidence interval (grey zone).", fig.height = 12, fig.width = 12, fig.align = "center"}


load("Mutation_rate/TFBS_exonic_mut_rates_facet.Rdata")
TFBS.exonic.mut.rate.facet

```





\pagebreak
```{r TFBS_flanks, cache=TRUE, include=TRUE, echo=FALSE, eval=TRUE, fig.cap="Mutation rates in TFBSs and their flanking regions. Left: ±50 bp, center: ±250 bp, right: ±500 bp.", fig.height = 9, fig.width = 7, fig.align = "center"}

load("Mutation_rate/Mut_rates_TFBS_vs_flanks_facet_all_cohorts.RData")
tfbs.vs.flanks.gg.all + add_space

```





\pagebreak
```{r Exon_flanks, cache=TRUE, include=TRUE, echo=FALSE, eval=TRUE, fig.cap="Mutation rates in exons and their flanking regions. Left: ±50 bp, center: ±250 bp, right: ±500 bp.", fig.height = 10, fig.width = 8, fig.align = "center"}

load("Mutation_rate/Mut_rates_Exons_vs_flanks_facet_all_cohorts.RData")
exons.vs.flanks.gg.all + add_space

```





\pagebreak
```{r GeneHancer_associations, cache=TRUE, include=TRUE, echo=FALSE, eval=TRUE, fig.cap="TFBS-gene associations. When a TFBS (purple box) lies within an GeneHancer-annotated cis-regulatory element (CRE; pink box), the TFBS is linked to its target genes according to GeneHancer, (see TFBS$_B$ and TFBS$_C$). When a TFBS is outside an annotated CRE (see TFBS$_A$, TFBS$_D$, and TFBS$_E$), it is associated with the closest TSS.", out.width = 500, fig.align = "center"}

knitr::include_graphics("GeneHancer/TFBS_CRE_association-1.png", dpi = 100)
```





\pagebreak
```{r xseq_model_explained, cache=TRUE, include=TRUE, echo=FALSE, eval=TRUE, fig.cap="Likelihood association of mutations with gene expression dysregulation in xseq. Taking advantage of the gene expression information (RNA-seq, microarrays), xseq identifies genes whose partners in a regulatory network have expression values deviated from neutral observed values, an enrichment of both upregulated and downregulated genes within a network is evaluated in individual samples and then across a cohort. Example with a network of target genes for a given miRNA. A) Gene regulatory status. For each target $i$ connected to the miRNA gene $X$ associated with a cis-regulatory mutation, xseq decomposes the expression distribution of $i$ (using all samples in the cohort) as a 3-component mixture model (down, neutral, and up). Next, xseq computes the posterior probability of observing the expression of $i$ in the mutated sample $S$ as belonging to the down, neutral, and up components. The component with the highest posterior probability provides information about the regulatory status of $i$ (down, neutral, or up). This step is repeated for each gene in the network and for each sample in the cohort. B) The sample-specific network dysregulation probability is calculated from the down and up regulation posterior probabilities of all the genes in the network for a given samples where the mutation was observed. C) The posterior probabilities of sample-specific dysregulation are used to compute the posterior probability of association between a cis-regulatory mutated miRNA and its target network dysregulation across a cohort. For a detailed explanation of the parameters and the graphical model, see the xseq publication by Ding et al., Nature Communications, 2015.", fig.align = "center", out.width = 500}

knitr::include_graphics("Model/Xseq_explained.png", dpi = 100)
```





\pagebreak
```{r Predicted_PCG_1, cache=TRUE, include=TRUE, echo=FALSE, eval=TRUE, fig.cap="Protein-coding genes predicted in BRCA-US, HNSC-US, LIHC-US, and LUSC-US TCGA cohorts. Predictions were obtained applying the xseq tool when considering protein-coding genes mutated through either LoF (red triangles) or cis-regulatory (TFBS; green triangles) mutations, independently. Cancer-associated genes (red stars) and TFs (blue stars) are highlighted and hypergeometric tests p-values for enrichment are provided in the legend (Material and methods).", out.width = "650pt", out.height = "550pt", fig.align = "center"}

knitr::include_graphics("Highlighted_genes_pancancer/xseq_trans_pancancer_heatmap_PCG_sep_by_cohort_P1.jpeg", dpi = 100)
```





\pagebreak
```{r Predicted_PCG_2, cache=TRUE, include=TRUE, echo=FALSE, eval=TRUE, fig.cap="Protein-coding genes predicted in LUAD-US, STAD-US, and UCEC-US TCGA cohorts. Predictions were obtained applying the xseq tool when considering protein-coding genes mutated through either LoF (red triangles) or cis-regulatory (TFBS; green triangles) mutations, independently. Cancer-associated genes (red stars) and TFs (blue stars) are highlighted and hypergeometric tests p-values for enrichment are provided in the legend (Material and methods).", out.width = "700pt", out.height="575pt", fig.align = "center"}

knitr::include_graphics("Highlighted_genes_pancancer/xseq_trans_pancancer_heatmap_PCG_sep_by_cohort_P2.jpeg", dpi = 100)
```





\pagebreak
```{r TFBS_vs_Exonic_mutations, cache=TRUE, include=TRUE, echo=FALSE, eval=TRUE, fig.cap="Protein-coding genes predicted through cis-regulatory mutations (i.e. within TFBSs) are generally not mutated in their exons in the same patients. A) For each cohort, number of samples where the predicted protein-coding genes through cis-regulatory mutations (TFBSs) also contain a LoF mutation. B) Same as in A) but providing the corresponding number of predicted protein-coding genes per cohort for each category.", fig.height = 15, fig.width = 13, fig.align = "center"}

load("Nb_TFBS_vs_Exonic_mutations/Analysis_samples_w_TFBS_Exonic_mutations.rdata")


## The original colors in the legend are inverted when the plots are reported as Rdata, therefore we have to reset the color scale
suppressMessages(barplots$A     <- barplots$A + scale_fill_manual(values = c(TFBS_and_Exonic = "#bf812d", TFBS = "#35978f"), labels = c("TFBS only", "TFBS & Exon")))

suppressMessages(barplots$extra <- barplots$extra + scale_fill_manual(values = c(TFBS_and_Exonic = "#bf812d", TFBS = "#35978f"), labels = c("TFBS only", "TFBS & Exon")))

suppressMessages(barplots$B     <- barplots$B + scale_fill_manual(values = c(TFBS_and_Exonic = "#bf812d", TFBS = "#35978f"), labels = c("TFBS only", "TFBS & Exon")))
  
font.size <- 100

tfbs.vs.exon.mutations <- plot_grid(barplots$A, barplots$extra, barplots$B,
                                    nrow = 3, rel_heights = c(1, 0.001, 1),
                                    labels = c("A)", "", "B)"),
                                    label_size = 25)
tfbs.vs.exon.mutations <- tfbs.vs.exon.mutations + add_space


suppressMessages(tfbs.vs.exon.mutations )

## NOTE: the variable barplots$C has the plot displayed in Figure 2B
## barplots$C 
```





\pagebreak
```{r Aggregated_networks_1, cache=TRUE, include=TRUE, echo=FALSE, eval=TRUE, fig.cap="Network of predicted genes in A) BRCA, B) HNSC, C) LIHC, D) LUAD, E) LUSC, and F) STAD cohorts, genes that could not be connected are not shown. The node colors indicate the number of samples where the gene was predicted. TFs genes are highlighted with orange background.", fig.height = 30, fig.width = 25, fig.align = "center", message=FALSE, warning=FALSE}

net.list      <- vector(mode = "list", length = length(cohorts))
nb.nodes.list <- vector(mode = "list", length = length(cohorts))

names(nb.nodes.list) <- cohorts

for (co in cohorts) {
  
  load(paste0("Aggregated_networks/", co, "/Aggregated_network_", co, "-US.Rdata"))
  agg.net <- agg.net.fun(df           = list.agg.net$DF,
                         project.name = list.agg.net$Project,
                         clusters     = list.agg.net$Clusters)  + add_space
  
  net.list[[co]] <- agg.net
  nb.nodes.list[[co]] <- list.agg.net$Clusters
  
}

## Mean/MEdian 
## 1) Cluster size
## 2) Nb of clusters
# summary(unlist(purrr::map(nb.nodes.list, 2)))
# summary(unlist(purrr::map(nb.nodes.list, 3)))




suppressMessages(plot_grid(net.list[["BRCA"]],
                           net.list[["HNSC"]],

                           net.list[["LIHC"]],
                           net.list[["LUAD"]],
                           
                           net.list[["LUSC"]],
                           net.list[["STAD"]],
                           
                           nrow = 3, labels = c('A)', 'B)', 'C)', 'D)', 'E)', 'F)'),
                           label_size = 35, vjust = 1, hjust = 0))

```




<!-- ```{r Figure3, cache=TRUE, include=TRUE, echo=FALSE, eval=TRUE, fig.cap="", ig.height = 18, fig.width = 15, fig.align = "center"} -->

<!--   cohorts <- c("BRCA", "HNSC", "LIHC", "LUAD", "LUSC", "STAD", "UCEC") -->

<!--   subgraph.list        <- vector(mode = "list", length = length(cohorts)) -->
<!--   names(subgraph.list) <- cohorts -->
<!--   for (co in cohorts) { -->

<!--     load(paste0("Aggregated_networks/", co, "/Aggregated_network_", co, "-US.Rdata")) -->

<!--     gene.cluster <- data.frame(list.agg.net$Clusters$membership) %>% -->
<!--                       rename(Cluster = list.agg.net.Clusters.membership) -->
<!--     gene.cluster$Gene   <- rownames(gene.cluster) -->
<!--     gene.cluster$Cohort <- paste0(co, "-US") -->

<!--     subgraph.list[[co]] <- gene.cluster -->

<!--   } -->


<!--   nb.subgraphs.per.cohort.df <- rbindlist(subgraph.list) %>% -->
<!--                                   group_by(Cohort, Cluster) %>% -->
<!--                                   summarise(N = n(), .groups = "drop") %>% -->
<!--                                   # ungroup() %>% -->
<!--                                   complete(Cluster, nesting(Cohort)) -->

<!--   nb.subgraphs.per.cohort.gg <- ggplot(nb.subgraphs.per.cohort.df, aes(y = Cluster, x = Cohort, fill = as.numeric(N))) + -->
<!--                                   geom_tile(aes(width = 0.95, height = 0.95)) + -->
<!--                                   geom_text(aes(label = N), size = 5) + -->
<!--                                   theme_classic() + -->
<!--                                   coord_equal() + -->
<!--                                   scale_x_discrete(limits = paste0(cohorts, "-US")) + -->
<!--                                   scale_y_discrete(limits = factor(1:12)) + -->
<!--                                   scale_fill_distiller(palette = "Oranges", direction = +1, na.value = "white", name = "Number of genes in subgraph") + -->
<!--                                   theme(plot.title         = element_text(hjust = .5), -->
<!--                                         plot.subtitle      = element_text(hjust = .5), -->
<!--                                         axis.text.y        = element_text(angle = 0, -->
<!--                                                                           hjust = 1, -->
<!--                                                                           size  = 15), -->
<!--                                         axis.text.x        = element_text(angle = 45, -->
<!--                                                                           hjust = 1, -->
<!--                                                                           size  = 15), -->
<!--                                         legend.position    = "bottom") + -->
<!--                                   labs(y = "Number of subgraphs", x = "") + -->
<!--                                   guides(fill = "none") -->


<!--   agg.nets.fig <- -->
<!--   plot_grid(net.list[["UCEC"]], -->
<!--             nb.subgraphs.per.cohort.gg, -->
<!--             nrow = 1, labels = c('A)', 'B)'), -->
<!--             align = "hv", -->
<!--             label_size = 20, vjust = 1.7, hjust = -1, rel_widths = c(1, 1)) -->

<!--   agg.nets.fig -->
<!--   # 11.5 * 6 -->


<!-- ``` -->




\pagebreak
```{r FEA_GO_BP, cache=TRUE, include=TRUE, echo=FALSE, eval=TRUE, fig.cap="Functional enrichment (GO Biological Process 2021) analysis considering dysregulated genes in the networks of predicted miRNA and protein-coding genes. The heatmaps represent the significance (-log$_{10}$(P-value)) of the most enriched terms (rows) associated to the dysregulated genes found on the seven analyzed cohorts (columns) when considering protein-coding genes with LoF mutations (PCGs::LoF; top-left), protein-coding genes with cis-regulatory mutations (PCG::TFBS; top-right), and miRNAs with cis-regulatory mutations (miRNAs::TFBS; bottom-right). The terms are ordered by their mean ranks across the cohorts.", fig.height = 110, fig.width = 97, fig.align = "center"}

types <- c("PCG::TFBS",
           "premiRNA::TFBS",
           "PCG::LoF")

load("FEA_heatmaps/GSEA_enrichment_pancancer_heatmap_ordered_labels.RData")  ## heatmap.GSEA.axis.list


db <- "GO_Biological_Process_2021"
font.size <- 70
# co <- "PCG::TFBS"
heatmap.GSEA.list <- list()
for(co in types){

  fea.df <- heatmap.GSEA.axis.list[[db]][[co]][["df"]]

  cohorts.sorted <- heatmap.GSEA.axis.list[[db]][[co]][["x_axis"]]
  terms.sorted <- heatmap.GSEA.axis.list[[db]][[co]][["y_axis"]]
  ti = co
  if(co=="premiRNA::TFBS")
    ti = "miRNA::TFBS"

  heatmap.GSEA.list[[co]] <-
    ggplot(fea.df, aes(x = Cohort, y = Term, fill = Significance)) +
        geom_tile(width = 0.95, height = 0.95) +
        # coord_equal() +
        theme_bw() +
        theme(legend.position = "bottom", legend.title=element_text(size=font.size)) +
        scale_fill_viridis_c(option = "A",direction = -1, breaks = floor(seq(0, max(fea.df$Significance), length.out = 4)), limits = c(0, max(fea.df$Significance))) + # limits = c(0, 75)
        # scale_fill_distiller(palette = "PuRd", direction = +1, breaks = gg.breaks) +
        theme(axis.text.x = element_text(angle = 45, hjust = 1, vjust=1, size=font.size, colour= "black"),
              axis.text.y = element_text(size=font.size, colour = "black"),
              axis.title.y = element_text(size=font.size,colour = "black",vjust=0.12),
              axis.title.x = element_text(size=font.size,colour = "black",vjust=0.12),
              legend.text=element_text(size=font.size),
              plot.title = element_text(size=font.size)) +
        scale_x_discrete(limits = cohorts.sorted) +
        scale_y_discrete(limits = terms.sorted) +
        labs(title = paste0(db, "\n", ti), x = "", y = "") +
        guides(fill = guide_colourbar(barwidth = 23, barheight = 6)) +
        guides(color = guide_legend(title.position = "top",
                                    title.hjust = 0.5,
                                    label.position = "left")) +
    ## Add a border to avoid trunked axis legends
      theme(
        panel.background = element_rect(fill = "white"),
        plot.margin = margin(0.25, 0.25, 0.25, 0.25, "cm"),
        plot.background = element_rect(
          fill = "#fefefe",
          colour = "white",
          size = 1))
}


enriched.terms.gg <- plot_grid(heatmap.GSEA.list[["PCG::LoF"]],
                               heatmap.GSEA.list[["PCG::TFBS"]], NULL,
                               heatmap.GSEA.list[["premiRNA::TFBS"]],
                               ncol = 2, align = "hv", labels = c('', '', ''), label_size = 30,
                               label_colour = "black", hjust = -2, vjust = 3,
                               rel_heights = c(1, 1, 1,1), rel_widths = c(1, 1, 1,1))

## Combine plots
enriched.terms.gg

rm(heatmap.GSEA.list)
rm(enriched.terms.gg)
```



\pagebreak
```{r FEA_KEGG_2021, cache=TRUE, include=TRUE, echo=FALSE, eval=TRUE, fig.cap="Functional enrichment (KEGG 2021) analysis considering dysregulated genes in the networks of predicted miRNA and protein-coding genes. The heatmaps represent the significance (-log$_{10}$(P-value)) of the most enriched terms (rows) associated to the dysregulated genes found on the seven analyzed cohorts (columns) when considering protein-coding genes with LoF mutations (PCGs::LoF; top-left), protein-coding genes with cis-regulatory mutations (PCG::TFBS; top-right), and miRNAs with cis-regulatory mutations (miRNAs::TFBS; bottom-right). The terms are ordered by their mean ranks across the cohorts.", fig.height = 70, fig.width = 60, fig.align = "center"}

# plot.list <- list()
# load("FEA_heatmaps/GSEA_enrichment_pancancer_heatmap.RData")  ## heatmap.GSEA.list
# plot.list[["PCG::TFBS"]] <- heatmap.GSEA.list[["KEGG_2019_Human"]][["PCG::TFBS"]]
# plot.list[["premiRNA::TFBS"]] <- heatmap.GSEA.list[["KEGG_2019_Human"]][["premiRNA::TFBS"]]
# plot.list[["PCG::LoF"]] <- heatmap.GSEA.list[["KEGG_2019_Human"]][["PCG::LoF"]]

types <- c("PCG::TFBS",
           "premiRNA::TFBS",
           "PCG::LoF")

load("FEA_heatmaps/GSEA_enrichment_pancancer_heatmap_ordered_labels.RData")  ## heatmap.GSEA.axis.list


db <- "KEGG_2021_Human"
font.size <- 55
# co <- "PCG::TFBS"

heatmap.GSEA.list <- list()
for(co in types){
  ti = co
  if(co=="premiRNA::TFBS")
    ti = "miRNA::TFBS"
  fea.df <- heatmap.GSEA.axis.list[[db]][[co]][["df"]]

  cohorts.sorted <- heatmap.GSEA.axis.list[[db]][[co]][["x_axis"]]
  terms.sorted <- heatmap.GSEA.axis.list[[db]][[co]][["y_axis"]]

  heatmap.GSEA.list[[co]] <-
    ggplot(fea.df, aes(x = Cohort, y = Term, fill = Significance)) +
        geom_tile(width = 0.95, height = 0.95) +
        # coord_equal() +
        theme_bw() +
        theme(legend.position = "bottom", legend.title=element_text(size=font.size)) +
        scale_fill_viridis_c(option = "A",direction = -1, breaks = floor(seq(0, max(fea.df$Significance), length.out = 4)), limits = c(0, max(fea.df$Significance))) + # limits = c(0, 75)
        # scale_fill_distiller(palette = "PuRd", direction = +1, breaks = gg.breaks) +
        theme(axis.text.x = element_text(angle = 45, hjust = 1, vjust=1, size=font.size, colour= "black"),
              axis.text.y = element_text(size=font.size, colour = "black"),
              axis.title.y = element_text(size=font.size,colour = "black",vjust=0.12),
              axis.title.x = element_text(size=font.size,colour = "black",vjust=0.12),
              legend.text=element_text(size=font.size),
              plot.title = element_text(size=font.size)) +
        scale_x_discrete(limits = cohorts.sorted) +
        scale_y_discrete(limits = terms.sorted) +
        labs(title = paste0(db, "\n", ti), x = "", y = "") +
        guides(fill = guide_colourbar(barwidth = 17, barheight = 6)) +
        guides(color = guide_legend(title.position = "top",
                                    title.hjust = 0.5,
                                    label.position = "left")) +
    ## Add a border to avoid trunked axis legends
      theme(
        panel.background = element_rect(fill = "white"),
        plot.margin = margin(0.25, 0.25, 0.25, 0.25, "cm"),
        plot.background = element_rect(
          fill = "#fefefe",
          colour = "white",
          size = 1))
}


enriched.terms.gg <- plot_grid(heatmap.GSEA.list[["PCG::LoF"]],
                               heatmap.GSEA.list[["PCG::TFBS"]], NULL,
                               heatmap.GSEA.list[["premiRNA::TFBS"]],
                               ncol = 2, align = "hv", labels = c('', '', ''), label_size = 30,
                               label_colour = "black", hjust = -2, vjust = 3,
                               rel_heights = c(1, 1, 1,1), rel_widths = c(1, 1, 1,1))

## Combine plots
enriched.terms.gg

rm(heatmap.GSEA.list)
rm(enriched.terms.gg)
```



<!-- ```{r FEA_KEGG_2021_Fig4_5, cache=TRUE, include=FALSE, echo=FALSE, eval=TRUE} -->

<!--   ## Fig 4 -->
<!--   ss <- plot_grid(heatmap.GSEA.list[["PCG::LoF"]], -->
<!--                   heatmap.GSEA.list[["PCG::TFBS"]], -->
<!--                   ncol = 2, align = "h", labels = c('', ''), label_size = 30, -->
<!--                   label_colour = "black", hjust = -2, vjust = 3, -->
<!--                   rel_heights = c(1, 1), rel_widths = c(0.85, 1)) -->

<!--   ggsave(ss, filename = "Main_manuscript/Fig4/KEGG_2021_PCG_LOF.pdf", width = 60, height = 40, limitsize = F) -->
<!--   ggsave(ss, filename = "Main_manuscript/Fig4/KEGG_2021_PCG_LOF.jpeg", width = 20, height = 18, limitsize = F) -->


<!--   ## Fig 5 -->

<!-- ``` -->



\pagebreak
```{r FEA_WIKIPATH, cache=TRUE, include=TRUE, echo=FALSE, eval=TRUE, fig.cap="Functional enrichment (WikiPathways 2019 human) analysis considering dysregulated genes in the networks of predicted miRNA and protein-coding genes. The heatmaps represent the significance (-log$_{10}$(P-value)) of the most enriched terms (rows) associated to the dysregulated genes found on the seven analyzed cohorts (columns) when considering protein-coding genes with LoF mutations (PCGs::LoF; top-left), protein-coding genes with cis-regulatory mutations (PCG::TFBS; top-right), and miRNAs with cis-regulatory mutations (miRNAs::TFBS; bottom-right). The terms are ordered by their mean ranks across the cohorts.", fig.height = 100, fig.width = 86, fig.align = "center"}

types <- c("PCG::TFBS",
           "premiRNA::TFBS",
           "PCG::LoF")

load("FEA_heatmaps/GSEA_enrichment_pancancer_heatmap_ordered_labels.RData")  ## heatmap.GSEA.axis.list


db <- "WikiPathways_2019_Human"
font.size <- 60
# co <- "PCG::TFBS"
heatmap.GSEA.list <- list()

for(co in types){
  ti = co
  if(co=="premiRNA::TFBS")
    ti = "miRNA::TFBS"
  fea.df <- heatmap.GSEA.axis.list[[db]][[co]][["df"]]

  cohorts.sorted <- heatmap.GSEA.axis.list[[db]][[co]][["x_axis"]]
  terms.sorted <- heatmap.GSEA.axis.list[[db]][[co]][["y_axis"]]

  heatmap.GSEA.list[[co]] <-
    ggplot(fea.df, aes(x = Cohort, y = Term, fill = Significance)) +
        geom_tile(width = 0.95, height = 0.95) +
        # coord_equal() +
        theme_bw() +
        theme(legend.position = "bottom", legend.title=element_text(size=font.size)) +
        scale_fill_viridis_c(option = "A",direction = -1, breaks = floor(seq(0, max(fea.df$Significance), length.out = 4)), limits = c(0, max(fea.df$Significance))) + # limits = c(0, 75)
        # scale_fill_distiller(palette = "PuRd", direction = +1, breaks = gg.breaks) +
        theme(axis.text.x = element_text(angle = 45, hjust = 1, vjust=1, size=font.size, colour= "black"),
              axis.text.y = element_text(size=font.size, colour = "black"),
              axis.title.y = element_text(size=font.size,colour = "black",vjust=0.12),
              axis.title.x = element_text(size=font.size,colour = "black",vjust=0.12),
              legend.text=element_text(size=font.size),
              plot.title = element_text(size=font.size)) +
        scale_x_discrete(limits = cohorts.sorted) +
        scale_y_discrete(limits = terms.sorted) +
        labs(title = paste0(db, "\n", ti), x = "", y = "") +
        guides(fill = guide_colourbar(barwidth = 17, barheight = 6)) +
        guides(color = guide_legend(title.position = "top",
                                    title.hjust = 0.5,
                                    label.position = "left")) +
    ## Add a border to avoid trunked axis legends
      theme(
        panel.background = element_rect(fill = "white"),
        plot.margin = margin(0.25, 0.25, 0.25, 0.25, "cm"),
        plot.background = element_rect(
          fill = "#fefefe",
          colour = "white",
          size = 1))
}


enriched.terms.gg <- plot_grid(heatmap.GSEA.list[["PCG::LoF"]],
                               heatmap.GSEA.list[["PCG::TFBS"]], NULL,
                               heatmap.GSEA.list[["premiRNA::TFBS"]],
                               ncol = 2, align = "hv", labels = c('', '', ''), label_size = 30,
                               label_colour = "black", hjust = -2, vjust = 3,
                               rel_heights = c(1, 1, 1,1), rel_widths = c(1, 1, 1,1))

## Combine plots
enriched.terms.gg

rm(heatmap.GSEA.list)
rm(enriched.terms.gg)
```



\pagebreak
```{r FEA_Panther, cache=TRUE, include=TRUE, echo=FALSE, eval=TRUE, fig.cap="Functional enrichment (Panther 2016) analysis considering dysregulated genes in the networks of predicted miRNA and protein-coding genes. The heatmaps represent the significance (-log$_{10}$(P-value)) of the most enriched terms (rows) associated to the dysregulated genes found on the seven analyzed cohorts (columns) when considering protein-coding genes with LoF mutations (PCGs::LoF; top-left), protein-coding genes with cis-regulatory mutations (PCG::TFBS; top-right), and miRNAs with cis-regulatory mutations (miRNAs::TFBS; bottom-right). The terms are ordered by their mean ranks across the cohorts.", fig.height = 87, fig.width = 77, fig.align = "center"}

types <- c("PCG::TFBS",
           "premiRNA::TFBS",
           "PCG::LoF")

load("FEA_heatmaps/GSEA_enrichment_pancancer_heatmap_ordered_labels.RData")  ## heatmap.GSEA.axis.list


db <- "Panther_2016"
font.size <- 65
# co <- "PCG::TFBS"
heatmap.GSEA.list <- list()

for(co in types){
  ti = co
  if(co=="premiRNA::TFBS")
    ti = "miRNA::TFBS"
  fea.df <- heatmap.GSEA.axis.list[[db]][[co]][["df"]]

  cohorts.sorted <- heatmap.GSEA.axis.list[[db]][[co]][["x_axis"]]
  terms.sorted <- heatmap.GSEA.axis.list[[db]][[co]][["y_axis"]]

  heatmap.GSEA.list[[co]] <-
    ggplot(fea.df, aes(x = Cohort, y = Term, fill = Significance)) +
        geom_tile(width = 0.95, height = 0.95) +
        # coord_equal() +
        theme_bw() +
        theme(legend.position = "bottom", legend.title=element_text(size=font.size)) +
        scale_fill_viridis_c(option = "A",direction = -1, breaks = floor(seq(0, max(fea.df$Significance), length.out = 4)), limits = c(0, max(fea.df$Significance))) + # limits = c(0, 75)
        # scale_fill_distiller(palette = "PuRd", direction = +1, breaks = gg.breaks) +
        theme(axis.text.x = element_text(angle = 45, hjust = 1, vjust=1, size=font.size, colour= "black"),
              axis.text.y = element_text(size=font.size, colour = "black"),
              axis.title.y = element_text(size=font.size,colour = "black",vjust=0.12),
              axis.title.x = element_text(size=font.size,colour = "black",vjust=0.12),
              legend.text=element_text(size=font.size),
              plot.title = element_text(size=font.size)) +
        scale_x_discrete(limits = cohorts.sorted) +
        scale_y_discrete(limits = terms.sorted) +
        labs(title = paste0(db, "\n", ti), x = "", y = "") +
        guides(fill = guide_colourbar(barwidth = 17, barheight = 6)) +
        guides(color = guide_legend(title.position = "top",
                                    title.hjust = 0.5,
                                    label.position = "left")) +
    ## Add a border to avoid trunked axis legends
      theme(
        panel.background = element_rect(fill = "white"),
        plot.margin = margin(0.25, 0.25, 0.25, 0.25, "cm"),
        plot.background = element_rect(
          fill = "#fefefe",
          colour = "white",
          size = 1))
}


enriched.terms.gg <- plot_grid(heatmap.GSEA.list[["PCG::LoF"]],
                               heatmap.GSEA.list[["PCG::TFBS"]], NULL,
                               heatmap.GSEA.list[["premiRNA::TFBS"]],
                               ncol = 2, align = "hv", labels = c('', '', ''), label_size = 30,
                               label_colour = "black", hjust = -2, vjust = 3,
                               rel_heights = c(1, 1, 1,1), rel_widths = c(1, 1, 1,1))

## Combine plots
enriched.terms.gg

rm(heatmap.GSEA.list)
rm(enriched.terms.gg)
```



\pagebreak
\begin{landscape}
```{r miRNAs_predicted_pancancer, cache=TRUE, include=TRUE, echo=FALSE, eval=TRUE, fig.cap="miRNAs predicted by xseq in the seven TCGA cohorts. Cell colors indicate the posterior probability computed over the corresponding cohort. Red stars indicate that the miRNA is annotated as a cancer-associated miRNA in miRCancer. Blue stars indicate that the miRNA was reported as a cancer-associated miRNA in the specific cancer type where it is predicted by xseq, according to miRCancer annotation.", out.width = "695pt", out.height = "700pt", fig.align = "center"}

## This plot is generated in this script:summary_xseq_trans_ICGC_primiRNA_genes_ALL_projects_Fig5.R
knitr::include_graphics("miRNA_pancancer/Mature_miRNAs_pancancer.jpeg", dpi = 100)
```
\end{landscape}



<!-- \pagebreak -->
<!-- ```{r Nb_target_genes_miRNAs, cache=TRUE, include=TRUE, echo=FALSE, eval=TRUE, fig.cap="miRNA-target genes information. A) Distribution of the number of targets considered per miRNA. The red line indicates the median value (1052 targets). The rug at the bottom highlights the predicted miRNAs. B) Number of cohorts (x-axis) where each mature miRNA was predicted versus its number of targets (y-axis).", fig.height = 10, fig.width = 10, fig.align = "center"} -->

<!-- plot.list <- list() -->
<!-- load("Nb_targets_miRNA/Nb_targets_distribution.RData")  ## nb.targets.distr -->
<!-- load("Nb_targets_miRNA/Nb_targets_vs_Nb_cohorts.RData") ## nb.targets.nb.cohorts.violin -->

<!-- plot.list[["Distrib"]] <- nb.targets.distr -->
<!-- plot.list[["Violin"]] <- nb.targets.nb.cohorts.violin -->


<!-- ## Add a border to avoid trunked axis legends -->
<!-- for(co in names(plot.list)){ -->

<!--   plot.list[[co]] <- -->
<!--     plot.list[[co]] + -->
<!--     theme( -->
<!--     panel.background = element_rect(fill = "white"), -->
<!--     plot.margin = margin(0.5, 0.5, 0.5, 0.5, "cm"), -->
<!--     plot.background = element_rect( -->
<!--       fill = "#fefefe", -->
<!--       colour = "white", -->
<!--       size = 1 -->
<!--     ) -->
<!--   ) -->
<!-- } -->

<!-- ## Combine plots -->
<!-- suppressMessages( -->
<!-- plot_grid(plot.list[["Distrib"]], -->
<!--           NULL, -->
<!--           plot.list[["Violin"]], -->
<!--           ncol = 1, labels = c('A)', '', 'B)'), label_size = 17, label_colour = "black", hjust = 0, vjust = 3, rel_heights = c(1, 0.15, 1)) -->
<!-- ) -->

<!-- rm(plot.list) -->
<!-- rm(nb.targets.distr) -->
<!-- rm(nb.targets.nb.cohorts.violin) -->
<!-- ``` -->



\pagebreak
```{r Fraction_dysreg_nets, cache=TRUE, include=TRUE, echo=FALSE, eval=TRUE, fig.cap="Fraction of the original gene networks predicted as dysregulated for the predicted genes in A) BRCA-US, B) HNSC-US, C) LIHC-US, D) LUAD-US, E) LUSC-US, F) STAT-US, and G) UCEC-US. X-axis corresponds to the number of dysregulated genes in the filtered network (see Material and methods), Y-axis corresponds to the original size of the gene networks before any filtering (for expression). Color scale corresponds to the fraction of the network that is predicted to be dysregulated in the samples with corresponding mutations of interest.", fig.height = 30, fig.width = 25, fig.align = "center"}

plot.list <- list()

## Load plots
load("Fraction_dysreg_net/Dysregulated_network_fraction_BRCA-US.Rdata") ## dysreg.network.fraction.plot
plot.list[["BRCA"]] <- dysreg.network.fraction.plot + theme_bw()

load("Fraction_dysreg_net/Dysregulated_network_fraction_HNSC-US.Rdata") ## dysreg.network.fraction.plot
plot.list[["HNSC"]] <- dysreg.network.fraction.plot + theme_bw()

load("Fraction_dysreg_net/Dysregulated_network_fraction_LIHC-US.Rdata") ## dysreg.network.fraction.plot
plot.list[["LIHC"]] <- dysreg.network.fraction.plot + theme_bw()

load("Fraction_dysreg_net/Dysregulated_network_fraction_LUAD-US.Rdata") ## dysreg.network.fraction.plot
plot.list[["LUAD"]] <- dysreg.network.fraction.plot + theme_bw()

load("Fraction_dysreg_net/Dysregulated_network_fraction_LUSC-US.Rdata") ## dysreg.network.fraction.plot
plot.list[["LUSC"]] <- dysreg.network.fraction.plot + theme_bw()

load("Fraction_dysreg_net/Dysregulated_network_fraction_STAD-US.Rdata") ## dysreg.network.fraction.plot
plot.list[["STAD"]] <- dysreg.network.fraction.plot + theme_bw()

load("Fraction_dysreg_net/Dysregulated_network_fraction_UCEC-US.Rdata") ## dysreg.network.fraction.plot
plot.list[["UCEC"]] <- dysreg.network.fraction.plot + theme_bw()


## Add a border to avoid trunked axis legends
for(co in names(plot.list)){

  plot.list[[co]] <-
    plot.list[[co]] +
    # scale_colour_gradient2(low = "#feb24c", mid = "#e31a1c", high = "#800026", midpoint = 0.5, limits = c(0,1)) +
    theme(
    panel.background = element_rect(fill = "white"),
    plot.margin = margin(0.75, 0.75, 0.75, 0.75, "cm"),
    plot.background = element_rect(
      fill = "#fefefe",
      colour = "white",
      size = 1
    )
  )
}


suppressWarnings(suppressMessages(
plot_grid(plot.list[["BRCA"]],
          plot.list[["HNSC"]],
          plot.list[["LIHC"]],
          plot.list[["LUAD"]],
          plot.list[["LUSC"]],
          plot.list[["STAD"]],
          plot.list[["UCEC"]],
          ncol = 2, labels = c('A)', 'B)', 'C)', 'D)', 'E)', 'F)', 'G)'), label_size = 35, label_colour = "black", hjust = 0, vjust = 1.15, rel_heights = c(1))
))


rm(plot.list)
rm(dysreg.network.fraction.plot)
```

                                
                                      
                                      
\pagebreak
```{r Predicted_genes_breast_cancer, cache=TRUE, include=TRUE, echo=FALSE, eval=TRUE, fig.cap="Predicted protein-coding genes in the ICGC cohort using all samples, ER+ samples and ER- samples, and in the TCGA BRCA-US cohort (columns). Predictions were obtained applying the xseq tool on each dataset independently considering protein-coding genes mutated through either LoF (red triangles) or cis-regulatory (TFBS; green triangles) mutations. Genes known as cancer genes (red stars) and TFs (blue stars) are highlighted and hypergeometric tests p-values for enrichment are provided in the legend (Material and methods).", out.width = "650pt", out.height="530pt", fig.align = "center"}

knitr::include_graphics("BASIS_vs_TCGA/BRCA_predicted_genes.jpeg", dpi = 100)
```



\begin{landscape}
\pagebreak
```{r FEA_breast_cancer_PCG, cache=TRUE, include=TRUE, echo=FALSE, eval=TRUE, fig.cap="Functional enrichment analysis considering dysregulated genes in the networks of predicted protein-coding genes in the breast cancer cohorts (columns). The heatmaps represent the significance (-log$_{10}$(P-value)) of the top-10 most enriched terms (rows) associated to the dysregulated genes for GO biological processes (GO-BP; A), WikiPathways (B), Panther (C), and KEGG (D). Terms (rows) are ordered by their mean rank across all cohorts.", fig.height = 75, fig.width = 115, fig.align = "center"}

## From: /storage/mathelierarea/processed/jamondra/Projects/dysmir/R_scripts/BASIS/Heatmap_enriched_terms_breast_cancer.R


load("BASIS_vs_TCGA/RData/GSEA_enrichment_pancancer_heatmap_ordered_labels.RData")  ## heatmap.GSEA.axis.list

db1 <- "KEGG_2021_Human"
db2 <- "WikiPathways_2019_Human"
db3 <- "Panther_2016"
db4 <- "GO_Biological_Process_2021"
DBs <- c(db1, db2, db3, db4)

db <- names(heatmap.GSEA.axis.list)
font.size <- 80
mt <- c("PCG.TFBS")
heatmap.GSEA.list <- list()
for(db in DBs){

  fea.df <- heatmap.GSEA.axis.list[[db]][[mt]][["df"]]

  cohorts.sorted <- heatmap.GSEA.axis.list[[db]][[mt]][["x_axis"]]
  terms.sorted <- heatmap.GSEA.axis.list[[db]][[mt]][["y_axis"]]

  db.title <- NULL
  if(db == "KEGG_2021_Human"){
    db.title <- "KEGG"
  }
  if(db == "WikiPathways_2019_Human"){
    db.title <- "WikiPathways"
    }
  if(db == "Panther_2016"){
    db.title <- "Panther"
  }
  if(db == "GO_Biological_Process_2021"){
    db.title <- "GO_BP"
  }

  heatmap.GSEA.list[[db]] <-
    ggplot(fea.df, aes(x = Cohort, y = Term, fill = Significance)) +
        geom_tile(width = 0.95, height = 0.95) +
        # coord_equal() +
        theme_bw() +
        theme(legend.position = "bottom", legend.title=element_text(size=font.size)) +
        scale_fill_viridis_c(option = "A",direction = -1, breaks = floor(seq(0, max(fea.df$Significance), length.out = 4)), limits = c(0, max(fea.df$Significance))) + # limits = c(0, 75)
        # scale_fill_distiller(palette = "PuRd", direction = +1, breaks = gg.breaks) +
        theme(axis.text.x = element_text(angle = 45, hjust = 1, vjust=1, size=font.size, colour= "black"),
              axis.text.y = element_text(size=font.size, colour = "black"),
              axis.title.y = element_text(size=font.size,colour = "black",vjust=0.12),
              axis.title.x = element_text(size=font.size,colour = "black",vjust=0.12),
              legend.text=element_text(size=font.size),
              plot.title = element_text(size=font.size+10)) +
        scale_x_discrete(limits = cohorts.sorted) +
        scale_y_discrete(limits = terms.sorted) +
        labs(title = db.title, x = "", y = "") +
        guides(fill = guide_colourbar(barwidth = 30, barheight = 7)) +
        guides(color = guide_legend(title.position = "top",
                                    title.hjust = 0.5,
                                    label.position = "left")) +
    ## Add a border to avoid trunked axis legends
      theme(
        panel.background = element_rect(fill = "white"),
        plot.margin = margin(0.5, 0.5, 0.5, 0.5, "cm"),
        plot.background = element_rect(
          fill = "#fefefe",
          colour = "white",
          size = 1))
}


# font.size <- 1
# null.hm <- ggplot(mtcars, aes(x=cyl)) +
#             geom_bar() +
#             theme(legend.position = "none")


enriched.terms.gg <- plot_grid(heatmap.GSEA.list[[db4]],
                               heatmap.GSEA.list[[db2]],
                               heatmap.GSEA.list[[db3]],
                               heatmap.GSEA.list[[db1]],
                               nrow = 2, align = "hv", labels = c('A)', 'B)', 'C)', 'D)'), label_size = 150,
                               label_colour = "black", hjust = -7, vjust = 2,
                               rel_heights = c(1, 1, 1,1), rel_widths = c(1, 1, 1,1))

## Combine plots
enriched.terms.gg

rm(heatmap.GSEA.list)
rm(enriched.terms.gg)
```
\end{landscape}


```{r Breast_Cancer_agg_net, cache=TRUE, include=TRUE, echo=FALSE, eval=TRUE, fig.cap="Network representing the predicted protein-coding genes in ICGC (all samples), ICGC ER+, ICGC ER- and BRCA-US cohorts. The name of the genes predicted in two or more cohorts are displayed many times with different colors.", fig.height = 14, fig.width = 18, fig.align = "center", message=FALSE, warning=FALSE}
# Taken from: ~/Documents/PostDoc/Mathelier_lab/Projects/R_utilities/R-scripts/Draw_network.R
# knitr::include_graphics("BASIS_vs_TCGA/BRCA_agg_net.jpeg", dpi = 100)
load("BASIS_vs_TCGA/RData/BRCA_BASIS_agg_net.RData")
# agg.net

## Figure 6 in the main manuscript
```



\pagebreak
\begin{landscape}
```{r miRNA_brca_ER_plot, cache=TRUE, include=TRUE, echo=FALSE, eval=TRUE, fig.cap="Predicted miRNA drivers in breast cancer cohorts. Predicted miRNAs in TCGA (BRCA-US) and ICGC (all samples and ER- samples only). The pie plots represent the distribution of samples with ER-/+ status.", out.width = "675pt", out.height = "900pt", fig.align = "center", message=FALSE, warning=FALSE}

knitr::include_graphics("BASIS_vs_TCGA/BRCA_miRNAs_pie.jpeg", dpi = 100)

## Taken from: /storage/mathelierarea/processed/jamondra/Projects/dysmir/BASIS/TEST/ER_MATCH/Heatmap_with_ER_status.R
```



\pagebreak
```{r FEA_breast_cancer_miRNAs, cache=TRUE, include=TRUE, echo=FALSE, eval=TRUE, fig.cap="Functional enrichment analysis considering dysregulated target genes of predicted miRNAs in the breast cancer cohorts (columns). The heatmaps represent the significance (-log$_{10}$(P-value)) of the top-10 most enriched terms (rows) associated to the dysregulated genes for GO biological processes (GO-BP; A) and WikiPathways (B). Terms (rows) are ordered by their mean rank across all cohorts.", fig.height = 70, fig.width = 115, fig.align = "center"}

## From: /storage/mathelierarea/processed/jamondra/Projects/dysmir/R_scripts/BASIS/Heatmap_enriched_terms_breast_cancer.R


load("BASIS_vs_TCGA/RData/GSEA_enrichment_pancancer_heatmap_ordered_labels.RData")  ## heatmap.GSEA.axis.list

## No KEGG for miRNAs
db1 <- "WikiPathways_2019_Human"
# db2 <- "Panther_2016"
db3 <- "GO_Biological_Process_2021"
DBs <- c(db1, db2, db3)

db <- names(heatmap.GSEA.axis.list)
font.size <- 80
mt <- c("premiRNA.TFBS")
heatmap.GSEA.list <- list()
for (db in DBs) {
  
  # message(db)

  fea.df <- heatmap.GSEA.axis.list[[db]][[mt]][["df"]]

  cohorts.sorted <- heatmap.GSEA.axis.list[[db]][[mt]][["x_axis"]]
  terms.sorted <- heatmap.GSEA.axis.list[[db]][[mt]][["y_axis"]]

  db.title <- NULL
  if (db == "KEGG_2021_Human") {
    db.title <- "KEGG"
  }
  if(db == "WikiPathways_2019_Human"){
    db.title <- "WikiPathways"
    }
  if(db == "Panther_2016"){
    db.title <- "Panther"
  }
  if(db == "GO_Biological_Process_2021"){
    db.title <- "GO_BP"
  }

  heatmap.GSEA.list[[db]] <-
    ggplot(fea.df, aes(x = Cohort, y = Term, fill = Significance)) +
        geom_tile(width = 0.95, height = 0.95) +
        # coord_equal() +
        theme_bw() +
        theme(legend.position = "bottom", legend.title=element_text(size=font.size)) +
        scale_fill_viridis_c(option = "A",direction = -1, breaks = floor(seq(0, max(fea.df$Significance), length.out = 4)), limits = c(0, max(fea.df$Significance))) + # limits = c(0, 75)
        # scale_fill_distiller(palette = "PuRd", direction = +1, breaks = gg.breaks) +
        theme(axis.text.x = element_text(angle = 45, hjust = 1, vjust=1, size=font.size, colour= "black"),
              axis.text.y = element_text(size=font.size, colour = "black"),
              axis.title.y = element_text(size=font.size,colour = "black",vjust=0.12),
              axis.title.x = element_text(size=font.size,colour = "black",vjust=0.12),
              legend.text=element_text(size=font.size),
              plot.title = element_text(size=font.size+10)) +
        scale_x_discrete(limits = cohorts.sorted) +
        scale_y_discrete(limits = terms.sorted) +
        labs(title = db.title, x = "", y = "") +
        guides(fill = guide_colourbar(barwidth = 30, barheight = 7)) +
        guides(color = guide_legend(title.position = "top",
                                    title.hjust = 0.5,
                                    label.position = "left")) +
    ## Add a border to avoid trunked axis legends
      theme(
        panel.background = element_rect(fill = "white"),
        plot.margin = margin(0.5, 0.5, 0.5, 0.5, "cm"),
        plot.background = element_rect(
          fill = "#fefefe",
          colour = "white",
          size = 1))
}

enriched.terms.gg <- plot_grid(heatmap.GSEA.list[[db3]],
                               heatmap.GSEA.list[[db1]],
                               # heatmap.GSEA.list[[db2]],
                               nrow = 2, align = "hv", labels = c('A)', 'B)'), label_size = 150,
                               label_colour = "black", hjust = -7, vjust = 1,
                               rel_heights = c(1, 1), rel_widths = c(1, 1))

## Combine plots
enriched.terms.gg

rm(heatmap.GSEA.list)
rm(enriched.terms.gg)
```
\end{landscape}



\pagebreak
```{r Survival_BCSS, cache=TRUE, include=TRUE, echo=FALSE, eval=TRUE, fig.cap="Breast cancer specific survival (BCSS, y-axis) plots for selected miRNAs. Kaplan-Meier survival curves were calculated using the METABRIC cohort. Samples were separated into two groups according to the level of miRNA expression (above/below the median).", fig.height = 30, fig.width = 25, fig.align = "center", warning = FALSE, message=FALSE}

# setwd("/home/jamondra/Documents/PostDoc/Mathelier_lab/Manuscript/Supp_Material")
load("Survival_plots/BCSS_survival_plots.RData") ## selected.mirnas.bcss.plot
# names(selected.mirnas.bcss.plot)


bcss.examples <-
suppressWarnings(
plot_grid(selected.mirnas.bcss.plot[["hsa-mir-10a-3p"]],
          selected.mirnas.bcss.plot[["hsa-mir-17-3p"]],
          selected.mirnas.bcss.plot[["hsa-mir-17-5p"]],
          # selected.mirnas.bcss.plot[["hsa-mir-18a-3p"]],
          selected.mirnas.bcss.plot[["hsa-mir-18a-5p"]],
          selected.mirnas.bcss.plot[["hsa-mir-20a-3p"]],
          selected.mirnas.bcss.plot[["hsa-mir-20a-5p"]],
          selected.mirnas.bcss.plot[["hsa-mir-21-3p"]],
          selected.mirnas.bcss.plot[["hsa-mir-21-5p"]],
          selected.mirnas.bcss.plot[["hsa-mir-29a-3p"]],
          selected.mirnas.bcss.plot[["hsa-mir-29a-5p"]],
          selected.mirnas.bcss.plot[["hsa-mir-92a-1-5p"]],
          selected.mirnas.bcss.plot[["hsa-mir-145-3p"]],
          selected.mirnas.bcss.plot[["hsa-mir-145-5p"]],
          selected.mirnas.bcss.plot[["hsa-mir-155-5p"]],
          selected.mirnas.bcss.plot[["hsa-mir-181c-3p"]],
          # selected.mirnas.bcss.plot[["hsa-mir-196b-3p"]],
          selected.mirnas.bcss.plot[["hsa-mir-205-3p"]],
          selected.mirnas.bcss.plot[["hsa-mir-205-5p"]],
          selected.mirnas.bcss.plot[["hsa-mir-339-5p"]],
          selected.mirnas.bcss.plot[["hsa-mir-342-3p"]],
          selected.mirnas.bcss.plot[["hsa-mir-342-5p"]],
          selected.mirnas.bcss.plot[["hsa-mir-378a-3p"]],
          # selected.mirnas.bcss.plot[["hsa-mir-508-3p"]],
          # selected.mirnas.bcss.plot[["hsa-mir-508-5p"]],
          selected.mirnas.bcss.plot[["hsa-mir-590-5p"]],
          selected.mirnas.bcss.plot[["hsa-mir-615-3p"]],
          # selected.mirnas.bcss.plot[["hsa-mir-615-5p"]],
          selected.mirnas.bcss.plot[["hsa-mir-629-3p"]],
          selected.mirnas.bcss.plot[["hsa-mir-708-3p"]],
          selected.mirnas.bcss.plot[["hsa-mir-708-5p"]],
          # selected.mirnas.bcss.plot[["hsa-mir-1247-3p"]],
          ncol = 5))

bcss.examples +
      theme(
    panel.background = element_rect(fill = NA, colour = "#fefefe"),
    plot.margin = margin(0.25, 0.25, 0.25, 0.25, "cm"),
    plot.background = element_rect(
      fill = NA,
      colour = NA,
      size = 1
    )
  )

```


\pagebreak
```{r Survival_OS, cache=TRUE, include=TRUE, echo=FALSE, eval=TRUE, fig.cap="Overall survival (OS, y-axis) plots for selected miRNAs. Kaplan-Meier survival curves were calculated using the METABRIC cohort. Samples were separated into two groups according to the level of miRNA expression (above/below the median).", fig.height = 32, fig.width = 25, fig.align = "center", warning = FALSE, message=FALSE}

## 
load("Survival_plots/OS_survival_plots.RData") ## selected.mirnas.os.plot

os.examples <-
suppressWarnings(
plot_grid(selected.mirnas.os.plot[["hsa-mir-10a-3p"]],
          selected.mirnas.os.plot[["hsa-mir-17-3p"]],
          selected.mirnas.os.plot[["hsa-mir-17-5p"]],
          # selected.mirnas.os.plot[["hsa-mir-18a-3p"]],
          selected.mirnas.os.plot[["hsa-mir-18a-5p"]],
          selected.mirnas.os.plot[["hsa-mir-20a-3p"]],
          selected.mirnas.os.plot[["hsa-mir-20a-5p"]],
          selected.mirnas.os.plot[["hsa-mir-21-3p"]],
          selected.mirnas.os.plot[["hsa-mir-21-5p"]],
          selected.mirnas.os.plot[["hsa-mir-29a-3p"]],
          selected.mirnas.os.plot[["hsa-mir-29a-5p"]],
          selected.mirnas.os.plot[["hsa-mir-92a-1-5p"]],
          selected.mirnas.os.plot[["hsa-mir-145-3p"]],
          selected.mirnas.os.plot[["hsa-mir-145-5p"]],
          selected.mirnas.os.plot[["hsa-mir-155-5p"]],
          selected.mirnas.os.plot[["hsa-mir-181c-3p"]],
          # selected.mirnas.os.plot[["hsa-mir-196b-3p"]],
          selected.mirnas.os.plot[["hsa-mir-205-3p"]],
          selected.mirnas.os.plot[["hsa-mir-205-5p"]],
          selected.mirnas.os.plot[["hsa-mir-339-5p"]],
          selected.mirnas.os.plot[["hsa-mir-342-3p"]],
          selected.mirnas.os.plot[["hsa-mir-342-5p"]],
          selected.mirnas.os.plot[["hsa-mir-378a-3p"]],
          # selected.mirnas.os.plot[["hsa-mir-508-3p"]],
          # selected.mirnas.os.plot[["hsa-mir-508-5p"]],
          selected.mirnas.os.plot[["hsa-mir-590-5p"]],
          selected.mirnas.os.plot[["hsa-mir-615-3p"]],
          # selected.mirnas.os.plot[["hsa-mir-615-5p"]],
          # selected.mirnas.os.plot[["hsa-mir-629-3p"]],
          selected.mirnas.os.plot[["hsa-mir-708-3p"]],
          selected.mirnas.os.plot[["hsa-mir-708-5p"]],
          selected.mirnas.os.plot[["hsa-mir-1247-3p"]],
          nrow = 6)
)

suppressWarnings(
os.examples +
      theme(
    panel.background = element_rect(fill = NA, colour = "#fefefe"),
    plot.margin = margin(0.25, 0.25, 0.25, 0.25, "cm"),
    plot.background = element_rect(
      fill = NA,
      colour = NA,
      size = 1
    )
  )
)
```



